<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Show features and regions of interest — focus • gggenomes</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  <script src="../extra.js"></script>

<meta property="og:title" content="Show features and regions of interest — focus" />
<meta property="og:description" content="Show loci containing features of interest. Loci can either be provided
as predefined regions directly (loci=), or are constructed automatically
based on pre-selected features (via ...). Features within max_dist are
greedily combined into the same locus. locate() adds these loci as new
track so that they can be easily visualized. focus() extracts those loci
from their parent sequences making them the new sequence set. These sequences
will have their locus_id as their new seq_id." />
<meta property="og:image" content="/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gggenomes.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes (old version)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thackl/gggenomes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Show features and regions of interest</h1>
    <small class="dont-index">Source: <a href='https://github.com/thackl/gggenomes/blob/master/R/focus.R'><code>R/focus.R</code></a></small>
    <div class="hidden name"><code>focus.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Show loci containing features of interest. Loci can either be provided
as predefined regions directly (<code>loci=</code>), or are constructed automatically
based on pre-selected features (via <code>...</code>). Features within <code>max_dist</code> are
greedily combined into the same locus. <code>locate()</code> adds these loci as new
track so that they can be easily visualized. <code>focus()</code> extracts those loci
from their parent sequences making them the new sequence set. These sequences
will have their <code>locus_id</code> as their new <code>seq_id</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>focus</span><span class='op'>(</span>
  <span class='va'>x</span>,
  <span class='va'>...</span>,
  .track_id <span class='op'>=</span> <span class='fl'>2</span>,
  .max_dist <span class='op'>=</span> <span class='fl'>10000</span>,
  .expand <span class='op'>=</span> <span class='fl'>5000</span>,
  .overhang <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"drop"</span>, <span class='st'>"trim"</span>, <span class='st'>"keep"</span><span class='op'>)</span>,
  .locus_id <span class='op'>=</span> <span class='fu'>str_glue</span><span class='op'>(</span><span class='st'>"{seq_id}_lc{row_number()}"</span><span class='op'>)</span>,
  .locus_id_group <span class='op'>=</span> <span class='va'>seq_id</span>,
  .locus_bin <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"bin"</span>, <span class='st'>"seq"</span>, <span class='st'>"locus"</span><span class='op'>)</span>,
  .locus_score <span class='op'>=</span> <span class='fu'>n</span><span class='op'>(</span><span class='op'>)</span>,
  .locus_filter <span class='op'>=</span> <span class='cn'>TRUE</span>,
  .loci <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span>

<span class='fu'>locate</span><span class='op'>(</span>
  <span class='va'>x</span>,
  <span class='va'>...</span>,
  .track_id <span class='op'>=</span> <span class='fl'>2</span>,
  .max_dist <span class='op'>=</span> <span class='fl'>10000</span>,
  .expand <span class='op'>=</span> <span class='fl'>5000</span>,
  .locus_id <span class='op'>=</span> <span class='fu'>str_glue</span><span class='op'>(</span><span class='st'>"{seq_id}_lc{row_number()}"</span><span class='op'>)</span>,
  .locus_id_group <span class='op'>=</span> <span class='va'>seq_id</span>,
  .locus_bin <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"bin"</span>, <span class='st'>"seq"</span>, <span class='st'>"locus"</span><span class='op'>)</span>,
  .locus_score <span class='op'>=</span> <span class='fu'>n</span><span class='op'>(</span><span class='op'>)</span>,
  .locus_filter <span class='op'>=</span> <span class='cn'>TRUE</span>,
  .locus_track <span class='op'>=</span> <span class='st'>"loci"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>A gggenomes object</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Logical predicates defined in terms of the variables in the track
given by <code>.track_id</code>. Multiple conditions are combined with ‘&amp;’. Only rows
where the condition evaluates to ‘TRUE’ are kept.</p>
<p>The arguments in ‘...’ are automatically quoted and evaluated in the
context of the data frame. They support unquoting and splicing. See
‘vignette("programming")’ for an introduction to these concepts.</p></td>
    </tr>
    <tr>
      <th>.track_id</th>
      <td><p>the track to filter from - defaults to first feature track,
usually "genes". Can be a quoted or unquoted string or a positional
argument giving the index of a track among all tracks (seqs, feats &amp;
links).</p></td>
    </tr>
    <tr>
      <th>.max_dist</th>
      <td><p>Maximum distance between adjacent features to be included
into the same locus, default 10kb.</p></td>
    </tr>
    <tr>
      <th>.expand</th>
      <td><p>The amount to nucleotides to expand the focus around the
target features. Default 2kb. Give two values for different up- and
downstream expansions.</p></td>
    </tr>
    <tr>
      <th>.overhang</th>
      <td><p>How to handle features overlapping the locus boundaries
(including expand). Options are to "keep" them, "trim" them exactly at the
boundaries, or "drop" all features not fully included within the
boundaries.</p></td>
    </tr>
    <tr>
      <th>.locus_id, .locus_id_group</th>
      <td><p>How to generate the ids for the new loci
which will eventually become their new <code>seq_id</code>s.</p></td>
    </tr>
    <tr>
      <th>.locus_bin</th>
      <td><p>What bin to assign new locus to. Defaults to keeping the
original binning, but can be set to the "seq" to bin all loci originating
from the same parent sequence, or to "locus" to separate all loci into
individual bins.</p></td>
    </tr>
    <tr>
      <th>.locus_score</th>
      <td><p>An expression evaluated in the context of all features
that are combined into a new locus. Results are stored in the column
<code>locus_score</code>. Defaults to the <code>n()</code>, i.e. the number of features per
locus. Set, for example, to <code><a href='https://rdrr.io/r/base/sum.html'>sum(bitscore)</a></code> to sum over all blast hit
bitscore of per locus. Usually used in conjunction with <code>.locus_filter</code>.</p></td>
    </tr>
    <tr>
      <th>.locus_filter</th>
      <td><p>An predicate expression used to post-filter identified
loci. Set <code>.locus_filter=locus_score &gt;= 3</code> to only return loci comprising
at least 3 target features.</p></td>
    </tr>
    <tr>
      <th>.loci</th>
      <td><p>A data.frame specifying loci directly. Required columns are
<code>seq_id,start,end</code>. Supersedes <code>...</code>.</p></td>
    </tr>
    <tr>
      <th>.locus_track</th>
      <td><p>The name of the new track containing the identified loci.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    
<ul>
<li><p><code>focus</code>: Identify regions of interest and zoom in on them</p></li>
<li><p><code>locate</code>: Identify regions of interest and add them as new feature track</p></li>
</ul>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'># Let's hunt some defense systems in marine SAGs</span>
<span class='co'># read the genomes</span>
<span class='va'>s0</span> <span class='op'>&lt;-</span> <span class='fu'><a href='read_tracks.html'>read_seqs</a></span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"gorg/gorg.fna"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'fasta' with `read_seq_len()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: gorg [/home/runner/work/_temp/Library/gggenomes/extdata/gorg/gorg.fna]</span></div><div class='input'><span class='va'>s1</span> <span class='op'>&lt;-</span> <span class='va'>s0</span> <span class='op'>%&gt;%</span>
  <span class='co'># strip trailing number from contigs to get bins</span>
  <span class='fu'>mutate</span><span class='op'>(</span>bin_id <span class='op'>=</span> <span class='fu'>str_remove</span><span class='op'>(</span><span class='va'>seq_id</span>, <span class='st'>"_\\d+$"</span><span class='op'>)</span><span class='op'>)</span>
<span class='co'># gene annotations from prokka</span>
<span class='va'>g0</span> <span class='op'>&lt;-</span> <span class='fu'><a href='read_tracks.html'>read_feats</a></span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"gorg/gorg.gff"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'gff3' with `read_gff3()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: gorg [/home/runner/work/_temp/Library/gggenomes/extdata/gorg/gorg.gff]</span></div><div class='output co'>#&gt; <span class='message'>Harmonizing column names</span>
#&gt; <span class='message'>* ID -&gt; feat_id</span>
#&gt; <span class='message'>* Parent -&gt; parent_ids</span>
#&gt; <span class='message'>* Name -&gt; name</span></div><div class='output co'>#&gt; <span class='message'>Features read</span></div><div class='output co'>#&gt; <span class='message'><span style='color: #949494;'># A tibble: 3 x 3</span></span>
#&gt; <span class='message'>  source       type      n</span>
#&gt; <span class='message'>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span></span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>1</span> Aragorn:1.2  tRNA     62</span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>2</span> barrnap:0.6  rRNA     10</span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>3</span> Prodigal:2.6 CDS    <span style='text-decoration: underline;'>3</span>268</span></div><div class='input'>
<span class='co'># best hits to the PADS Arsenal database of prokaryotic defense-system genes</span>
<span class='co'># $ mmseqs easy-search gorg.fna pads-arsenal-v1-prf gorg-pads-defense.o6 /tmp \</span>
<span class='co'>#     --greedy-best-hits</span>
<span class='va'>f0</span> <span class='op'>&lt;-</span> <span class='fu'><a href='read_tracks.html'>read_feats</a></span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"gorg/gorg-pads-defense.o6"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'blast' with `read_blast()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: gorg-pads-defense [/home/runner/work/_temp/Library/gggenomes/extdata/gorg/gorg-pads-defense.o6]</span></div><div class='input'><span class='va'>f1</span> <span class='op'>&lt;-</span> <span class='va'>f0</span> <span class='op'>%&gt;%</span>
  <span class='co'># parser system/gene info</span>
  <span class='fu'>separate</span><span class='op'>(</span><span class='va'>seq_id2</span>, into<span class='op'>=</span><span class='fu'>qc</span><span class='op'>(</span><span class='va'>seq_id2</span>, <span class='va'>system</span>, <span class='va'>gene</span><span class='op'>)</span>, sep<span class='op'>=</span><span class='st'>","</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span>
    <span class='va'>evalue</span> <span class='op'>&lt;</span> <span class='fl'>1e-10</span>,  <span class='co'># get rid of some spurious hits</span>
    <span class='co'># and let's focus just on a few systems for this example</span>
    <span class='va'>system</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"CRISPR-CAS"</span>, <span class='st'>"DISARM"</span>, <span class='st'>"GABIJA"</span>, <span class='st'>"LAMASSU"</span>, <span class='st'>"THOERIS"</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># plot the distribution of hits across full genomes</span>
<span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span><span class='va'>g0</span>, <span class='va'>s1</span>, <span class='va'>f1</span>, wrap<span class='op'>=</span><span class='fl'>2e5</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_color_brewer</span><span class='op'>(</span>palette<span class='op'>=</span><span class='st'>"Dark2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_point</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>x<span class='op'>=</span><span class='va'>x</span>,y<span class='op'>=</span><span class='va'>y</span>, color<span class='op'>=</span><span class='va'>system</span><span class='op'>)</span>, data<span class='op'>=</span><span class='fu'><a href='feats.html'>feats</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='img'><img src='focus-1.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># hilight the regions containing hits</span>
<span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span><span class='va'>g0</span>, <span class='va'>s1</span>, <span class='va'>f1</span>, wrap<span class='op'>=</span><span class='fl'>2e5</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>locate</span><span class='op'>(</span>.track_id <span class='op'>=</span> <span class='va'>feats</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/base/identity.html'>identity</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_color_brewer</span><span class='op'>(</span>palette<span class='op'>=</span><span class='st'>"Dark2"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_feat.html'>geom_feat</a></span><span class='op'>(</span>data<span class='op'>=</span><span class='fu'><a href='feats.html'>feats</a></span><span class='op'>(</span><span class='va'>loci</span><span class='op'>)</span>, color<span class='op'>=</span><span class='st'>"plum3"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_point</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>x<span class='op'>=</span><span class='va'>x</span>,y<span class='op'>=</span><span class='va'>y</span>, color<span class='op'>=</span><span class='va'>system</span><span class='op'>)</span>, data<span class='op'>=</span><span class='fu'><a href='feats.html'>feats</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Adding 'loci' track. Plot with `geom_feat(data=feats(loci))`</span></div><div class='img'><img src='focus-2.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># zoom in on loci</span>
<span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span><span class='va'>g0</span>, <span class='va'>s1</span>, <span class='va'>f1</span>, wrap<span class='op'>=</span><span class='fl'>5e4</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>focus</span><span class='op'>(</span>.track_id <span class='op'>=</span> <span class='va'>feats</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_gene.html'>geom_gene</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_feat.html'>geom_feat</a></span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>color<span class='op'>=</span><span class='va'>system</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_feat_text.html'>geom_feat_tag</a></span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>label<span class='op'>=</span><span class='va'>gene</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_color_brewer</span><span class='op'>(</span>palette<span class='op'>=</span><span class='st'>"Dark2"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Showing 36 loci with the following size distribution</span>
#&gt; <span class='message'>* min: 2500</span>
#&gt; <span class='message'>* q25: 6464</span>
#&gt; <span class='message'>* med: 9739</span>
#&gt; <span class='message'>* q75: 10714</span>
#&gt; <span class='message'>* max: 19561</span></div><div class='img'><img src='focus-3.png' alt='' width='960' height='480' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Thomas Hackl, Markus J. Ankenbrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


