<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Draw gene models — geom_gene • gggenomes</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  <script src="../extra.js"></script>

<meta property="og:title" content="Draw gene models — geom_gene" />
<meta property="og:description" content="Draw coding sequences, mRNAs and other non-coding features. Supports
multi-exon features. CDS and mRNAs in the same group are plotted together.
They can therefore also be positioned as a single unit using the position
argument." />
<meta property="og:image" content="/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.4.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gggenomes.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes (old version)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thackl/gggenomes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Draw gene models</h1>
    <small class="dont-index">Source: <a href='https://github.com/thackl/gggenomes/blob/master/R/geom_gene.R'><code>R/geom_gene.R</code></a></small>
    <div class="hidden name"><code>geom_gene.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Draw coding sequences, mRNAs and other non-coding features. Supports
multi-exon features. CDS and mRNAs in the same group are plotted together.
They can therefore also be positioned as a single unit using the <code>position</code>
argument.</p>
    </div>

    <pre class="usage"><span class='fu'>geom_gene</span><span class='op'>(</span>
  mapping <span class='op'>=</span> <span class='cn'>NULL</span>,
  data <span class='op'>=</span> <span class='fu'><a href='feats.html'>genes</a></span><span class='op'>(</span><span class='op'>)</span>,
  stat <span class='op'>=</span> <span class='st'>"identity"</span>,
  position <span class='op'>=</span> <span class='st'>"identity"</span>,
  na.rm <span class='op'>=</span> <span class='cn'>FALSE</span>,
  show.legend <span class='op'>=</span> <span class='cn'>NA</span>,
  inherit.aes <span class='op'>=</span> <span class='cn'>TRUE</span>,
  size <span class='op'>=</span> <span class='fl'>2</span>,
  rna_size <span class='op'>=</span> <span class='va'>size</span>,
  shape <span class='op'>=</span> <span class='va'>size</span>,
  rna_shape <span class='op'>=</span> <span class='va'>shape</span>,
  intron_shape <span class='op'>=</span> <span class='va'>size</span>,
  intron_types <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"CDS"</span>, <span class='st'>"mRNA"</span>, <span class='st'>"tRNA"</span>, <span class='st'>"tmRNA"</span>, <span class='st'>"ncRNA"</span>, <span class='st'>"rRNA"</span><span class='op'>)</span>,
  cds_aes <span class='op'>=</span> <span class='cn'>NULL</span>,
  rna_aes <span class='op'>=</span> <span class='cn'>NULL</span>,
  intron_aes <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>mapping</th>
      <td><p>Set of aesthetic mappings created by <code><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes()</a></code> or
<code><a href='https://ggplot2.tidyverse.org/reference/aes_.html'>aes_()</a></code>. If specified and <code>inherit.aes = TRUE</code> (the
default), it is combined with the default mapping at the top level of the
plot. You must supply <code>mapping</code> if there is no plot mapping.</p></td>
    </tr>
    <tr>
      <th>data</th>
      <td><p>The data to be displayed in this layer. There are three
options:</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot()</a></code>.</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href='https://ggplot2.tidyverse.org/reference/fortify.html'>fortify()</a></code> for which variables will be created.</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p></td>
    </tr>
    <tr>
      <th>stat</th>
      <td><p>The statistical transformation to use on the data for this
layer, as a string.</p></td>
    </tr>
    <tr>
      <th>position</th>
      <td><p>Position adjustment, either as a string, or the result of
a call to a position adjustment function.</p></td>
    </tr>
    <tr>
      <th>show.legend</th>
      <td><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p></td>
    </tr>
    <tr>
      <th>inherit.aes</th>
      <td><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href='https://ggplot2.tidyverse.org/reference/borders.html'>borders()</a></code>.</p></td>
    </tr>
    <tr>
      <th>size, rna_size</th>
      <td><p>the size of the gene model, aka the height of the
polygons. <code>rna_size</code> only applies to non-coding parts of the gene model,
defaults to size.</p></td>
    </tr>
    <tr>
      <th>shape, rna_shape</th>
      <td><p>vector of height and width of the arrow tip, defaults
to size. If only one value is provided it is recycled. Set '0' to
deactivates arrow-shaped tips. <code>rna_shape</code> only applies to non-coding parts
of the gene model, defaults to shape.</p></td>
    </tr>
    <tr>
      <th>intron_shape</th>
      <td><p>single value controlling the kink of the intron line.
Defaults to size. Set 0 for straight lines between exons.</p></td>
    </tr>
    <tr>
      <th>intron_types</th>
      <td><p>introns will only be computed/drawn for features with
types listed here. Set to "CDS" to plot mRNAs as continous features, and
set to NA to completely ignore introns.</p></td>
    </tr>
    <tr>
      <th>cds_aes, rna_aes, intron_aes</th>
      <td><p>overwrite aesthetics for different model
parts. Need to be wrapped in <code><a href='https://ggplot2.tidyverse.org/reference/aes.html'>ggplot2::aes()</a></code>. NOTE: These remappings are
applied after the data has been transformed and mapped by the plot scales
(see <code><a href='https://ggplot2.tidyverse.org/reference/aes_eval.html'>ggplot2::after_scale()</a></code>). So you need to map between aesthetic names
(not data columns) and with standardized names, i.e. British English
spelling. These mappings can be used to dynamically change parts of the
gene model. For example, to change the color of introns from a hard-coded
"black" to the same color used to fill the CDS you could specify
<code>intron_aes=aes(colour = fill)</code>. By default, <code>rna_aes</code> is remapped with
<code>aes(fill=colorspace::lighten(fill, .5), colour=colorspace::lighten(colour, .5))</code> to give it a lighter appearence than the corresponding CDS but in the
same color.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="aesthetics"><a class="anchor" href="#aesthetics"></a>Aesthetics</h2>

    

<p><code>geom_gene()</code> understands the following aesthetics (required aesthetics are in bold):</p><ul>
<li><p><strong><code>x</code></strong></p></li>
<li><p><strong><code>xend</code></strong></p></li>
<li><p><strong><code>y</code></strong></p></li>
<li><p><code>alpha</code></p></li>
<li><p><code>colour</code></p></li>
<li><p><code>fill</code></p></li>
<li><p><code>group</code></p></li>
<li><p><code>introns</code></p></li>
<li><p><code>linetype</code></p></li>
<li><p><code>stroke</code></p></li>
<li><p><code>type</code></p></li>
</ul><p>Learn more about setting these aesthetics in <code>vignette("ggplot2-specs")</code>.</p>


<p>'type' and 'group' (mapped to 'type' and 'geom_id' by default) power the
proper recognition of CDS and their corresponding mRNAs so that they can be
drawn as one composite object. Overwrite 'group' to plot CDS and mRNAs
independently.</p>
<p>'introns' (mapped to 'introns') is used to compute intron/exon boundaries.
Use the parameter <code>intron_types</code> if you want to disable introns.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>genes<span class='op'>=</span><span class='va'>emale_genes</span><span class='op'>)</span> <span class='op'>+</span>
 <span class='fu'>geom_gene</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>No seqs provided, inferring seqs from feats</span></div><div class='img'><img src='geom_gene-1.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>genes<span class='op'>=</span><span class='va'>emale_genes</span><span class='op'>)</span> <span class='op'>+</span>
 <span class='fu'>geom_gene</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/numeric.html'>as.numeric</a></span><span class='op'>(</span><span class='va'>gc_content</span><span class='op'>)</span><span class='op'>)</span>,position<span class='op'>=</span><span class='st'>"strand"</span><span class='op'>)</span> <span class='op'>+</span>
 <span class='fu'>scale_fill_viridis_b</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>No seqs provided, inferring seqs from feats</span></div><div class='img'><img src='geom_gene-2.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='va'>types</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='cn'>NA</span>, <span class='st'>"CDS"</span>, <span class='st'>"mRNA"</span>, <span class='st'>"tRNA"</span>, <span class='st'>"tmRNA"</span>, <span class='st'>"ncRNA"</span>, <span class='st'>"rRNA"</span>, <span class='st'>"intron"</span>,
 <span class='st'>"misc_RNA"</span>, <span class='st'>"mobile_element"</span>, <span class='st'>"operon"</span>, <span class='st'>"..."</span><span class='op'>)</span>
<span class='va'>g0</span> <span class='op'>&lt;-</span> <span class='fu'>tibble</span><span class='op'>(</span>seq_id<span class='op'>=</span><span class='st'>"A"</span>, start<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_along</a></span><span class='op'>(</span><span class='va'>types</span><span class='op'>)</span><span class='op'>*</span><span class='fl'>10</span>,
 end<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_along</a></span><span class='op'>(</span><span class='va'>types</span><span class='op'>)</span><span class='op'>*</span><span class='fl'>10</span><span class='op'>+</span><span class='fl'>7</span>, type<span class='op'>=</span><span class='va'>types</span>, introns<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3</span>,<span class='fl'>5</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>genes<span class='op'>=</span><span class='va'>g0</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># all features in the "genes" regardless of type</span>
  <span class='fu'><a href='geom_feat.html'>geom_feat</a></span><span class='op'>(</span>data<span class='op'>=</span><span class='fu'><a href='feats.html'>feats</a></span><span class='op'>(</span><span class='va'>genes</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_text</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>label<span class='op'>=</span><span class='st'>"geom_feat"</span>, x<span class='op'>=</span><span class='op'>-</span><span class='fl'>15</span>, y<span class='op'>=</span><span class='fl'>.9</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://rdrr.io/r/graphics/plot.window.html'>xlim</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>20</span>, <span class='cn'>NA</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># only features in the "genes" of geneish type (implicit `data=genes()`)</span>
  <span class='fu'>geom_gene</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_feat_text.html'>geom_gene_tag</a></span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>label<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>type</span><span class='op'>)</span>, <span class='st'>"&lt;NA&gt;"</span>, <span class='va'>type</span><span class='op'>)</span><span class='op'>)</span>, data<span class='op'>=</span><span class='fu'><a href='feats.html'>genes</a></span><span class='op'>(</span>.gene_types <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_text</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>label<span class='op'>=</span><span class='st'>"geom_gene"</span>, x<span class='op'>=</span><span class='op'>-</span><span class='fl'>15</span>, y<span class='op'>=</span><span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># control which types are returned from the track</span>
  <span class='fu'>geom_gene</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>y<span class='op'>=</span><span class='fl'>1.1</span><span class='op'>)</span>, data<span class='op'>=</span><span class='fu'><a href='feats.html'>genes</a></span><span class='op'>(</span>.gene_types <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"CDS"</span>, <span class='st'>"misc_RNA"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_text</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>label<span class='op'>=</span><span class='st'>"gene_types"</span>, x<span class='op'>=</span><span class='op'>-</span><span class='fl'>15</span>, y<span class='op'>=</span><span class='fl'>1.1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># control which types can have introns</span>
  <span class='fu'>geom_gene</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>y<span class='op'>=</span><span class='fl'>1.2</span>, yend<span class='op'>=</span><span class='fl'>1.2</span><span class='op'>)</span>, data<span class='op'>=</span><span class='fu'><a href='feats.html'>genes</a></span><span class='op'>(</span>.gene_types <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"CDS"</span>, <span class='st'>"misc_RNA"</span><span class='op'>)</span><span class='op'>)</span>,  intron_types <span class='op'>=</span> <span class='st'>"misc_RNA"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_text</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>label<span class='op'>=</span><span class='st'>"intron_types"</span>, x<span class='op'>=</span><span class='op'>-</span><span class='fl'>15</span>, y<span class='op'>=</span><span class='fl'>1.2</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>No seqs provided, inferring seqs from feats</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Ignoring unknown aesthetics: yend</span></div><div class='img'><img src='geom_gene-3.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># spliced genes</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span>
<span class='va'>g0</span> <span class='op'>&lt;-</span> <span class='fu'><a href='read_gff3.html'>read_gff3</a></span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"eden-utr.gff"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Harmonizing attribute names</span>
#&gt; <span class='message'><span style='color: #00BBBB;'>•</span> ID -&gt; feat_id</span>
#&gt; <span class='message'><span style='color: #00BBBB;'>•</span> Name -&gt; name</span>
#&gt; <span class='message'><span style='color: #00BBBB;'>•</span> Parent -&gt; parent_ids</span>
#&gt; <span class='message'><span style='color: #00BBBB;'>•</span> Target -&gt; target</span></div><div class='output co'>#&gt; <span class='message'>Features read</span></div><div class='output co'>#&gt; <span class='message'><span style='color: #949494;'># A tibble: 8 × 3</span></span>
#&gt; <span class='message'>  source type                n</span>
#&gt; <span class='message'>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>           <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span></span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>1</span> <span style='color: #BB0000;'>NA</span>     cDNA_match          1</span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>2</span> <span style='color: #BB0000;'>NA</span>     CDS                 5</span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>3</span> <span style='color: #BB0000;'>NA</span>     exon                5</span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>4</span> <span style='color: #BB0000;'>NA</span>     five_prime_UTR      1</span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>5</span> <span style='color: #BB0000;'>NA</span>     gene                1</span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>6</span> <span style='color: #BB0000;'>NA</span>     mRNA                5</span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>7</span> <span style='color: #BB0000;'>NA</span>     TF_binding_site     1</span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>8</span> <span style='color: #BB0000;'>NA</span>     three_prime_UTR     1</span></div><div class='input'><span class='va'>gg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>genes<span class='op'>=</span><span class='va'>g0</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>No seqs provided, inferring seqs from feats</span></div><div class='input'><span class='va'>gg</span> <span class='op'>+</span> <span class='fu'>geom_gene</span><span class='op'>(</span>position<span class='op'>=</span><span class='st'>"pile"</span><span class='op'>)</span> <span class='op'>+</span>
<span class='va'>gg</span> <span class='op'>+</span> <span class='fu'>geom_gene</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='va'>type</span><span class='op'>)</span>, position<span class='op'>=</span><span class='st'>"pile"</span>,
         shape <span class='op'>=</span> <span class='fl'>0</span>, intron_shape <span class='op'>=</span> <span class='fl'>0</span>, color<span class='op'>=</span><span class='st'>"white"</span><span class='op'>)</span> <span class='op'>+</span>
<span class='co'># some fine-control on cds/rna/intron after_scale aesthetics</span>
<span class='va'>gg</span> <span class='op'>+</span> <span class='fu'>geom_gene</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='va'>geom_id</span><span class='op'>)</span>, position<span class='op'>=</span><span class='st'>"pile"</span>,
         size <span class='op'>=</span> <span class='fl'>2</span>, shape <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>4</span>,<span class='fl'>3</span><span class='op'>)</span>, rna_size <span class='op'>=</span> <span class='fl'>2</span>, intron_shape <span class='op'>=</span> <span class='fl'>4</span>, stroke<span class='op'>=</span><span class='fl'>0</span>,
         cds_aes<span class='op'>=</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='st'>"black"</span><span class='op'>)</span>, rna_aes<span class='op'>=</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='va'>fill</span><span class='op'>)</span>,
         intron_aes<span class='op'>=</span><span class='fu'>aes</span><span class='op'>(</span>colour<span class='op'>=</span><span class='va'>fill</span>, stroke<span class='op'>=</span><span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
     <span class='fu'>scale_fill_viridis_d</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
<span class='co'># fun with introns</span>
<span class='va'>gg</span> <span class='op'>+</span> <span class='fu'>geom_gene</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='va'>geom_id</span><span class='op'>)</span>, position<span class='op'>=</span><span class='st'>"pile"</span>, size <span class='op'>=</span> <span class='fl'>3</span>, shape<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>4</span>,<span class='fl'>4</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
<span class='va'>gg</span> <span class='op'>+</span> <span class='fu'>geom_gene</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='va'>geom_id</span><span class='op'>)</span>, position<span class='op'>=</span><span class='st'>"pile"</span>, size <span class='op'>=</span> <span class='fl'>3</span>, shape<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>4</span>,<span class='fl'>4</span><span class='op'>)</span>,
         intron_types<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
<span class='va'>gg</span> <span class='op'>+</span> <span class='fu'>geom_gene</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='va'>geom_id</span><span class='op'>)</span>, position<span class='op'>=</span><span class='st'>"pile"</span>, size <span class='op'>=</span> <span class='fl'>3</span>, shape<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>4</span>,<span class='fl'>4</span><span class='op'>)</span>,
         intron_types<span class='op'>=</span><span class='st'>"CDS"</span><span class='op'>)</span>
</div><div class='img'><img src='geom_gene-4.png' alt='' width='960' height='480' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Thomas Hackl, Markus J. Ankenbrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


