<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot genomes, features and synteny maps — gggenomes • gggenomes</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  <script src="../extra.js"></script>

<meta property="og:title" content="Plot genomes, features and synteny maps — gggenomes" />
<meta property="og:description" content="Plot genomes, features and synteny maps" />
<meta property="og:image" content="/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.3.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gggenomes.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes (old version)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thackl/gggenomes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot genomes, features and synteny maps</h1>
    <small class="dont-index">Source: <a href='https://github.com/thackl/gggenomes/blob/master/R/gggenomes.R'><code>R/gggenomes.R</code></a></small>
    <div class="hidden name"><code>gggenomes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot genomes, features and synteny maps</p>
    </div>

    <pre class="usage"><span class='fu'>gggenomes</span><span class='op'>(</span>
  genes <span class='op'>=</span> <span class='cn'>NULL</span>,
  seqs <span class='op'>=</span> <span class='cn'>NULL</span>,
  feats <span class='op'>=</span> <span class='cn'>NULL</span>,
  links <span class='op'>=</span> <span class='cn'>NULL</span>,
  .id <span class='op'>=</span> <span class='st'>"file_id"</span>,
  spacing <span class='op'>=</span> <span class='fl'>0.05</span>,
  wrap <span class='op'>=</span> <span class='cn'>NULL</span>,
  infer_bin_id <span class='op'>=</span> <span class='va'>seq_id</span>,
  infer_start <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>min</a></span><span class='op'>(</span><span class='va'>start</span>, <span class='va'>end</span><span class='op'>)</span>,
  infer_end <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='va'>start</span>, <span class='va'>end</span><span class='op'>)</span>,
  infer_length <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='va'>start</span>, <span class='va'>end</span><span class='op'>)</span>,
  theme <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"clean"</span>, <span class='cn'>NULL</span><span class='op'>)</span>,
  .layout <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>genes, feats</th>
      <td><p>A data.frame, a list of data.frames, or a character vector
with paths to files containing gene data. Each item is added as feature
track.</p>
<p>For a single data.frame the track_id will be "genes" and "feats",
respectively. For a list, track_ids are parsed from the list names, or if
names are missing from the name of the variable containing each data.frame.
Data columns:</p><ul>
<li><p>required: <code>seq_id,start,end</code></p></li>
<li><p>recognized: <code>strand,bin_id,feat_id,introns</code></p></li>
</ul></td>
    </tr>
    <tr>
      <th>seqs</th>
      <td><p>A data.frame or a character vector with paths to files containing
sequence data. Data columns:</p><ul>
<li><p>required: <code>seq_id,length</code></p></li>
<li><p>recognized: <code>bin_id,start,end,strand</code></p></li>
</ul></td>
    </tr>
    <tr>
      <th>links</th>
      <td><p>A data.frame or a character vector with paths to files
containing link data. Each item is added as links track. Data columns:</p><ul>
<li><p>required: <code>seq_id,seq_id2</code></p></li>
<li><p>recognized: <code>start,end,bin_id,start2,end2,bin_id2,strand</code></p></li>
</ul></td>
    </tr>
    <tr>
      <th>.id</th>
      <td><p>The name of the column for file labels that are created when
reading directly from files. Defaults to "file_id". Set to "bin_id" if
every file represents a different bin.</p></td>
    </tr>
    <tr>
      <th>spacing</th>
      <td><p>between sequences in bases (&gt;1) or relative to longest bin (&lt;1)</p></td>
    </tr>
    <tr>
      <th>wrap</th>
      <td><p>wrap bins into multiple lines with at most this many nucleotides
per lin.</p></td>
    </tr>
    <tr>
      <th>infer_length, infer_start, infer_end, infer_bin_id</th>
      <td><p>used to infer pseudo
seqs if only feats or links are provided, or if no bin_id column was
provided. The expressions are evaluated in the context of the first feat
or link track.</p>
<p>By default subregions of sequences from the first to the last feat/link
are generated. Set <code>infer_start</code> to 0 to show all sequences from their
true beginning.</p></td>
    </tr>
    <tr>
      <th>theme</th>
      <td><p>choose a gggenomes default theme, NULL to omit.</p></td>
    </tr>
    <tr>
      <th>.layout</th>
      <td><p>a pre-computed layout from <code><a href='layout_genomes.html'>layout_genomes()</a></code>. Useful for
developmental purposes.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>gggenomes-flavored ggplot object</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Compare the genomic organization of three viral elements</span>
<span class='co'># EMALEs: endogenous mavirus-like elements (example data shipped with gggenomes)</span>
<span class='fu'>gggenomes</span><span class='op'>(</span><span class='va'>emale_genes</span>, <span class='va'>emale_seqs</span>, <span class='va'>emale_tirs</span>, <span class='va'>emale_ava</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>                  <span class='co'># chromosomes and labels</span>
  <span class='fu'><a href='geom_feat.html'>geom_feat</a></span><span class='op'>(</span>size<span class='op'>=</span><span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span>                              <span class='co'># terminal inverted repeats</span>
  <span class='fu'><a href='geom_gene.html'>geom_gene</a></span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='va'>strand</span><span class='op'>)</span>, position<span class='op'>=</span><span class='st'>"strand"</span><span class='op'>)</span> <span class='op'>+</span> <span class='co'># genes</span>
  <span class='fu'><a href='geom_link.html'>geom_link</a></span><span class='op'>(</span>offset <span class='op'>=</span> <span class='fl'>0.15</span><span class='op'>)</span>                         <span class='co'># synteny-blocks</span>
</div><div class='img'><img src='gggenomes-1.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># with some more information</span>
<span class='fu'>gggenomes</span><span class='op'>(</span><span class='va'>emale_genes</span>, <span class='va'>emale_seqs</span>, <span class='va'>emale_tirs</span>, <span class='va'>emale_ava</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='add_feats.html'>add_feats</a></span><span class='op'>(</span><span class='va'>emale_ngaros</span>, <span class='va'>emale_gc</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='add_feats.html'>add_clusters</a></span><span class='op'>(</span><span class='va'>emale_cogs</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='flip.html'>flip_by_links</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_link.html'>geom_link</a></span><span class='op'>(</span>offset <span class='op'>=</span> <span class='fl'>0.15</span>, color<span class='op'>=</span><span class='st'>"white"</span><span class='op'>)</span> <span class='op'>+</span>                        <span class='co'># synteny-blocks</span>
  <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>                                  <span class='co'># chromosomes and labels</span>
  <span class='co'># thistle4, salmon4, burlywood4</span>
  <span class='fu'><a href='geom_feat.html'>geom_feat</a></span><span class='op'>(</span>size<span class='op'>=</span><span class='fl'>6</span>, position<span class='op'>=</span><span class='st'>"identity"</span><span class='op'>)</span> <span class='op'>+</span>                         <span class='co'># terminal inverted repeats</span>
  <span class='fu'><a href='geom_feat.html'>geom_feat</a></span><span class='op'>(</span>data<span class='op'>=</span><span class='fu'><a href='feats.html'>feats</a></span><span class='op'>(</span><span class='va'>emale_ngaros</span><span class='op'>)</span>, color<span class='op'>=</span><span class='st'>"turquoise4"</span>, alpha<span class='op'>=</span><span class='fl'>.3</span>,
            position<span class='op'>=</span><span class='st'>"strand"</span>, size<span class='op'>=</span><span class='fl'>16</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_feat_text.html'>geom_feat_note</a></span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>label<span class='op'>=</span><span class='va'>type</span><span class='op'>)</span>, data<span class='op'>=</span><span class='fu'><a href='feats.html'>feats</a></span><span class='op'>(</span><span class='va'>emale_ngaros</span><span class='op'>)</span>,
                 position<span class='op'>=</span><span class='st'>"strand"</span>, nudge_y <span class='op'>=</span> <span class='fl'>.3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_gene.html'>geom_gene</a></span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>fill<span class='op'>=</span><span class='va'>cluster_id</span><span class='op'>)</span>, position<span class='op'>=</span><span class='st'>"strand"</span><span class='op'>)</span> <span class='op'>+</span> <span class='co'># genes</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span><span class='st'>"Conserved genes"</span>, palette<span class='op'>=</span><span class='st'>"Dark2"</span>, na.value <span class='op'>=</span> <span class='st'>"cornsilk3"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'>#scale_fill_viridis_b() +</span>
  <span class='fu'>geom_ribbon</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>x<span class='op'>=</span><span class='op'>(</span><span class='va'>x</span><span class='op'>+</span><span class='va'>xend</span><span class='op'>)</span><span class='op'>/</span><span class='fl'>2</span>, ymax<span class='op'>=</span><span class='va'>y</span><span class='op'>+</span><span class='fl'>.24</span>, ymin<span class='op'>=</span><span class='va'>y</span><span class='op'>+</span><span class='fl'>.38</span><span class='op'>-</span><span class='op'>(</span><span class='fl'>.4</span><span class='op'>*</span><span class='va'>score</span><span class='op'>)</span>,
                  group<span class='op'>=</span><span class='va'>seq_id</span>, linetype<span class='op'>=</span><span class='st'>"GC-content"</span><span class='op'>)</span>, <span class='fu'><a href='feats.html'>feats</a></span><span class='op'>(</span><span class='va'>emale_gc</span><span class='op'>)</span>,
              fill<span class='op'>=</span><span class='st'>"lavenderblush4"</span>, position<span class='op'>=</span><span class='fu'>position_nudge</span><span class='op'>(</span>y<span class='op'>=</span><span class='op'>-</span><span class='fl'>.1</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Joining, by = "feat_id"</span></div><div class='output co'>#&gt; <span class='message'>Flipping: 2,3,4</span></div><div class='img'><img src='gggenomes-2.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># initialize plot directly from files</span>
<span class='fu'>gggenomes</span><span class='op'>(</span>
  <span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.gff"</span><span class='op'>)</span>,
  <span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.gff"</span><span class='op'>)</span>,
  <span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales-tirs.gff"</span><span class='op'>)</span>,
  <span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.paf"</span><span class='op'>)</span>
<span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_gene.html'>geom_gene</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_feat.html'>geom_feat</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_link.html'>geom_link</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'gff3' with `read_gff3()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.gff]</span></div><div class='output co'>#&gt; <span class='message'>Harmonizing attribute names</span>
#&gt; <span class='message'>* ID -&gt; feat_id</span>
#&gt; <span class='message'>* Name -&gt; name</span>
#&gt; <span class='message'>* Note -&gt; note</span></div><div class='output co'>#&gt; <span class='message'>Features read</span></div><div class='output co'>#&gt; <span class='message'><span style='color: #949494;'># A tibble: 1 × 3</span></span>
#&gt; <span class='message'>  source type      n</span>
#&gt; <span class='message'>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span></span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>1</span> MFG    CDS     143</span></div><div class='output co'>#&gt; <span class='message'>Reading 'gff3' with `read_gff3()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales-tirs [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales-tirs.gff]</span></div><div class='output co'>#&gt; <span class='message'>Harmonizing attribute names</span>
#&gt; <span class='message'>* ID -&gt; feat_id</span>
#&gt; <span class='message'>* Name -&gt; name</span></div><div class='output co'>#&gt; <span class='message'>Features read</span></div><div class='output co'>#&gt; <span class='message'><span style='color: #949494;'># A tibble: 1 × 3</span></span>
#&gt; <span class='message'>  source type              n</span>
#&gt; <span class='message'>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>         <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span></span>
#&gt; <span class='message'><span style='color: #BCBCBC;'>1</span> MFG    repeat_region    12</span></div><div class='output co'>#&gt; <span class='message'>Reading 'paf' with `read_paf()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.paf]</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Found tags in max_tags column, you should increase max_tags to</span></div><div class='output co'>#&gt; <span class='message'>Read and tidied up a .paf file with 10 optional tag fields:</span>
#&gt; <span class='message'>NM, ms, AS, nn, tp, cm, s1, de, rl, cg</span>
#&gt; <span class='message'>Note: warnings about fewer than expected columns are expected for this format.</span></div><div class='output co'>#&gt; <span class='message'>Reading 'gff3' with `read_seq_len()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.gff]</span></div><div class='img'><img src='gggenomes-3.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># multi-contig genomes wrap to fixed width</span>
<span class='va'>s0</span> <span class='op'>&lt;-</span> <span class='fu'><a href='read_tracks.html'>read_seqs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.files.html'>list.files</a></span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"cafeteria"</span><span class='op'>)</span>, <span class='st'>"Cr.*\\.fa"</span>, full.names <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'fasta' with `read_seq_len()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: CrBVI [/home/runner/work/_temp/Library/gggenomes/extdata/cafeteria/CrBVI.fa]</span></div><div class='output co'>#&gt; <span class='message'>* file_id: CrCflag [/home/runner/work/_temp/Library/gggenomes/extdata/cafeteria/CrCflag.fa]</span></div><div class='output co'>#&gt; <span class='message'>* file_id: CrE410P [/home/runner/work/_temp/Library/gggenomes/extdata/cafeteria/CrE410P.fa]</span></div><div class='output co'>#&gt; <span class='message'>* file_id: CrRCC970 [/home/runner/work/_temp/Library/gggenomes/extdata/cafeteria/CrRCC970.fa]</span></div><div class='input'><span class='va'>s1</span> <span class='op'>&lt;-</span> <span class='va'>s0</span> <span class='op'>%&gt;%</span> <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>length</span> <span class='op'>&gt;</span> <span class='fl'>5e5</span><span class='op'>)</span>
<span class='fu'>gggenomes</span><span class='op'>(</span>seqs<span class='op'>=</span><span class='va'>s1</span>, infer_bin_id<span class='op'>=</span><span class='va'>file_id</span>, wrap<span class='op'>=</span><span class='fl'>5e6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_seq_label.html'>geom_seq_label</a></span><span class='op'>(</span><span class='op'>)</span>
</div><div class='img'><img src='gggenomes-4.png' alt='' width='960' height='480' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Thomas Hackl, Markus J. Ankenbrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


