<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Pick bins and seqs by name or position — pick • gggenomes</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  <script src="../extra.js"></script>

<meta property="og:title" content="Pick bins and seqs by name or position — pick" />
<meta property="og:description" content="Pick which bins and seqs to show and in what order. Uses
dplyr::select()-like syntax, which means unquoted genome names, positional
arguments and selection helpers, such as
tidyselect::starts_with() are supported. Renaming is not supported." />
<meta property="og:image" content="/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gggenomes.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes (old version)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thackl/gggenomes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Pick bins and seqs by name or position</h1>
    <small class="dont-index">Source: <a href='https://github.com/thackl/gggenomes/blob/master/R/pick.R'><code>R/pick.R</code></a></small>
    <div class="hidden name"><code>pick.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Pick which bins and seqs to show and in what order. Uses
<code><a href='https://dplyr.tidyverse.org/reference/select.html'>dplyr::select()</a></code>-like syntax, which means unquoted genome names, positional
arguments and <a href='https://tidyselect.r-lib.org/reference/language.html'>selection helpers</a>, such as
<code><a href='https://tidyselect.r-lib.org/reference/starts_with.html'>tidyselect::starts_with()</a></code> are supported. Renaming is not supported.</p>
    </div>

    <pre class="usage"><span class='fu'>pick</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='fu'>pick_seqs</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>...</span>, .bins <span class='op'>=</span> <span class='fu'>everything</span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'>pick_seqs_within</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>...</span>, .bins <span class='op'>=</span> <span class='fu'>everything</span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'>pick_by_tree</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>tree</span>, infer_bin_id <span class='op'>=</span> <span class='va'>label</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>...</th>
      <td><p>bins/seqs to pick, select-like expression.</p></td>
    </tr>
    <tr>
      <th>.bins</th>
      <td><p>scope for positional arguments, select-like expression, enclose
multiple arguments with <code><a href='https://rdrr.io/r/base/c.html'>c()</a></code>!</p></td>
    </tr>
    <tr>
      <th>tree</th>
      <td><p>a phylogenetic tree in ggtree or ape-"phylo" format.</p></td>
    </tr>
    <tr>
      <th>infer_bin_id</th>
      <td><p>an expression to extract bin_ids from the tree data.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Use the dots to select bins or sequences (depending on function suffix), and
the <code>.bins</code> argument to set the scope for positional arguments. For example,
<code>pick_seqs(1)</code> will pick the first sequence from the first bin, while
<code>pick_seqs(1, .bins=3)</code> will pick the first sequence from the third bin.</p>
    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    
<ul>
<li><p><code>pick</code>: pick bins by bin_id, positional argument (start at top)
or select-helper.</p></li>
<li><p><code>pick_seqs</code>: pick individual seqs seq_id, positional argument (start at
top left) or select-helper.</p></li>
<li><p><code>pick_seqs_within</code>: pick individual seqs but only modify bins containing those
seqs, keep rest as is.</p></li>
<li><p><code>pick_by_tree</code>: align bins with the leaves in a given phylogenetic tree.</p></li>
</ul>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>s0</span> <span class='op'>&lt;-</span> <span class='fu'>tibble</span><span class='op'>(</span>
bin_id <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"A"</span>, <span class='st'>"B"</span>, <span class='st'>"B"</span>, <span class='st'>"B"</span>, <span class='st'>"C"</span>, <span class='st'>"C"</span>, <span class='st'>"C"</span><span class='op'>)</span>,
seq_id <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"a1"</span>,<span class='st'>"b1"</span>,<span class='st'>"b2"</span>,<span class='st'>"b3"</span>,<span class='st'>"c1"</span>,<span class='st'>"c2"</span>,<span class='st'>"c3"</span><span class='op'>)</span>,
length <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1e4</span>, <span class='fl'>6e3</span>, <span class='fl'>2e3</span>, <span class='fl'>1e3</span>, <span class='fl'>3e3</span>, <span class='fl'>3e3</span>, <span class='fl'>3e3</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>p</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>seqs<span class='op'>=</span><span class='va'>s0</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>color<span class='op'>=</span><span class='va'>bin_id</span><span class='op'>)</span>, size<span class='op'>=</span><span class='fl'>3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_seq_label.html'>geom_seq_label</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>expand_limits</span><span class='op'>(</span>color<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"A"</span>,<span class='st'>"B"</span>,<span class='st'>"C"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>p</span>
</div><div class='img'><img src='pick-1.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># remove</span>
<span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick</span><span class='op'>(</span><span class='op'>-</span><span class='va'>B</span><span class='op'>)</span>
</div><div class='img'><img src='pick-2.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># select and reorder, by ID and position</span>
<span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick</span><span class='op'>(</span><span class='va'>C</span>, <span class='fl'>1</span><span class='op'>)</span>
</div><div class='img'><img src='pick-3.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># use helper function</span>
<span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick</span><span class='op'>(</span><span class='fu'>starts_with</span><span class='op'>(</span><span class='st'>"B"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='img'><img src='pick-4.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># pick just some seqs</span>
<span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick_seqs</span><span class='op'>(</span><span class='fl'>1</span>, <span class='va'>c3</span><span class='op'>)</span>
</div><div class='img'><img src='pick-5.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># pick with .bin scope</span>
<span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick_seqs</span><span class='op'>(</span><span class='fl'>3</span><span class='op'>:</span><span class='fl'>1</span>, .bins<span class='op'>=</span><span class='va'>C</span><span class='op'>)</span>
</div><div class='img'><img src='pick-6.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># change seqs in some bins, but keep rest as is</span>
<span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick_seqs_within</span><span class='op'>(</span><span class='fl'>3</span><span class='op'>:</span><span class='fl'>1</span>, .bins<span class='op'>=</span><span class='va'>B</span><span class='op'>)</span>
</div><div class='img'><img src='pick-7.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># same w/o scope, unaffected bins remain as is</span>
<span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick_seqs_within</span><span class='op'>(</span><span class='va'>b3</span>, <span class='va'>b2</span>, <span class='va'>b1</span><span class='op'>)</span>
</div><div class='img'><img src='pick-8.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># Align sequences with and plot next to a phylogenetic tree</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span>  <span class='co'># arrange multiple plots</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://yulab-smu.top/treedata-book/'>ggtree</a></span><span class='op'>)</span>     <span class='co'># plot phylogenetic trees</span>
</div><div class='output co'>#&gt; <span class='message'>ggtree v3.1.1  For help: https://yulab-smu.top/treedata-book/</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>If you use ggtree in published research, please cite the most appropriate paper(s):</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>1. Guangchuang Yu. Using ggtree to visualize data on tree-like structures. Current Protocols in Bioinformatics, 2020, 69:e96. doi:10.1002/cpbi.96</span>
#&gt; <span class='message'>2. Guangchuang Yu, Tommy Tsan-Yuk Lam, Huachen Zhu, Yi Guan. Two methods for mapping and visualizing associated data on phylogeny using ggtree. Molecular Biology and Evolution 2018, 35(12):3041-3043. doi:10.1093/molbev/msy194</span>
#&gt; <span class='message'>3. Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution 2017, 8(1):28-36. doi:10.1111/2041-210X.12628</span>
#&gt; <span class='message'></span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘ggtree’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:gggenomes’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    flip</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:tidyr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand</span></div><div class='input'>
<span class='co'># load and plot a phylogenetic tree</span>
<span class='va'>emale_mcp_tree</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/ape/man/read.tree.html'>read.tree</a></span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales-MCP.nwk"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>t</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/ggtree/man/ggtree.html'>ggtree</a></span><span class='op'>(</span><span class='va'>emale_mcp_tree</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://rdrr.io/pkg/ggtree/man/geom_tiplab.html'>geom_tiplab</a></span><span class='op'>(</span>align<span class='op'>=</span><span class='cn'>T</span>, size<span class='op'>=</span><span class='fl'>3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/lims.html'>xlim</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='fl'>0.05</span><span class='op'>)</span> <span class='co'># make room for labels</span>

<span class='va'>p</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>seqs<span class='op'>=</span><span class='va'>emale_seqs</span>, genes<span class='op'>=</span><span class='va'>emale_genes</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span>

<span class='co'># plot next to each other, but with</span>
<span class='co'># different order in tree and genomes</span>
<span class='va'>t</span> <span class='op'>+</span> <span class='va'>p</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>widths <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>5</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='img'><img src='pick-9.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># reorder genomes to match tree order</span>
<span class='co'># with a warning caused by mismatch in y-scale expansions</span>
<span class='va'>t</span> <span class='op'>+</span> <span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick_by_tree</span><span class='op'>(</span><span class='va'>t</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>widths <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>5</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: Tree and genomes have different y-scale expansions. This can cause slight misalignments of leaves and sequences.</span>
#&gt; <span class='warning'>Consider adding `+ scale_y_continuous(expand=c(0.01,0.7,0.01,0.7))` to the tree as a fix</span>
#&gt; <span class='warning'>* tree: 0,0.6,0,0.6</span>
#&gt; <span class='warning'>* bins: 0.01,0.7,0.01,0.7</span></div><div class='img'><img src='pick-10.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='co'># extra genomes are dropped with a notification</span>
<span class='va'>emale_seqs_more</span> <span class='op'>&lt;-</span> <span class='va'>emale_seqs</span>
<span class='va'>emale_seqs_more</span><span class='op'>[</span><span class='fl'>7</span>,<span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>emale_seqs_more</span><span class='op'>[</span><span class='fl'>6</span>,<span class='op'>]</span>
<span class='va'>emale_seqs_more</span><span class='op'>$</span><span class='va'>seq_id</span><span class='op'>[</span><span class='fl'>7</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='st'>"One more genome"</span>
<span class='va'>p</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>seqs<span class='op'>=</span><span class='va'>emale_seqs_more</span>, genes<span class='op'>=</span><span class='va'>emale_genes</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='va'>t</span> <span class='op'>+</span> <span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick_by_tree</span><span class='op'>(</span><span class='va'>t</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>widths <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>5</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Some bin_ids are missing in the tree, will drop those from genomes.</span>
#&gt; <span class='message'>* One more genome</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Tree and genomes have different y-scale expansions. This can cause slight misalignments of leaves and sequences.</span>
#&gt; <span class='warning'>Consider adding `+ scale_y_continuous(expand=c(0.01,0.7,0.01,0.7))` to the tree as a fix</span>
#&gt; <span class='warning'>* tree: 0,0.6,0,0.6</span>
#&gt; <span class='warning'>* bins: 0.01,0.7,0.01,0.7</span></div><div class='img'><img src='pick-11.png' alt='' width='960' height='480' /></div><div class='input'>
<span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='co'># no shared ids will cause an error</span>
  <span class='va'>p</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>seqs<span class='op'>=</span><span class='fu'>tibble</span><span class='op'>(</span>seq_id <span class='op'>=</span> <span class='st'>"foo"</span>, length<span class='op'>=</span><span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span>
  <span class='va'>t</span> <span class='op'>+</span> <span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick_by_tree</span><span class='op'>(</span><span class='va'>t</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>widths <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>5</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># extra leafs in tree will cause an error</span>
  <span class='va'>emale_seqs_fewer</span> <span class='op'>&lt;-</span> <span class='fu'>slice_head</span><span class='op'>(</span><span class='va'>emale_seqs</span>, n<span class='op'>=</span><span class='fl'>4</span><span class='op'>)</span>
  <span class='va'>p</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>seqs<span class='op'>=</span><span class='va'>emale_seqs_fewer</span>, genes<span class='op'>=</span><span class='va'>emale_genes</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='geom_bin_label.html'>geom_bin_label</a></span><span class='op'>(</span><span class='op'>)</span>
  <span class='va'>t</span> <span class='op'>+</span> <span class='va'>p</span> <span class='op'>%&gt;%</span> <span class='fu'>pick_by_tree</span><span class='op'>(</span><span class='va'>t</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>widths <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>5</span><span class='op'>)</span><span class='op'>)</span>
<span class='op'>}</span>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Thomas Hackl, Markus J. Ankenbrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


