<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Use tracks inside and outside geom_* calls — feats • gggenomes</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  <script src="../extra.js"></script>

<meta property="og:title" content="Use tracks inside and outside geom_* calls — feats" />
<meta property="og:description" content="Track selection works like dplyr::pull() and supports unquoted ids and
positional arguments. ... can be used to subset the data in
dplyr::filter() fashion. pull-prefixed variants return the specified
track from a gggenome object. Unprefixed variants work inside geom_* calls." />
<meta property="og:image" content="/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.3.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gggenomes.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes (old version)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thackl/gggenomes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Use tracks inside and outside <code>geom_*</code> calls</h1>
    <small class="dont-index">Source: <a href='https://github.com/thackl/gggenomes/blob/master/R/pull_tracks.R'><code>R/pull_tracks.R</code></a></small>
    <div class="hidden name"><code>pull_track.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Track selection works like <code><a href='https://dplyr.tidyverse.org/reference/pull.html'>dplyr::pull()</a></code> and supports unquoted ids and
positional arguments. <code>...</code> can be used to subset the data in
<code><a href='https://dplyr.tidyverse.org/reference/filter.html'>dplyr::filter()</a></code> fashion. <code>pull</code>-prefixed variants return the specified
track from a gggenome object. Unprefixed variants work inside <code>geom_*</code> calls.</p>
    </div>

    <pre class="usage"><span class='fu'>feats</span><span class='op'>(</span>.track_id <span class='op'>=</span> <span class='fl'>1</span>, <span class='va'>...</span>, .ignore <span class='op'>=</span> <span class='st'>"genes"</span>, .geneify <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='fu'>genes</span><span class='op'>(</span><span class='va'>...</span>, .gene_types <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"CDS"</span>, <span class='st'>"mRNA"</span>, <span class='st'>"tRNA"</span>, <span class='st'>"tmRNA"</span>, <span class='st'>"ncRNA"</span>, <span class='st'>"rRNA"</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'>links</span><span class='op'>(</span>.track_id <span class='op'>=</span> <span class='fl'>1</span>, <span class='va'>...</span>, .ignore <span class='op'>=</span> <span class='cn'>NULL</span>, .adjacent_only <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='fu'>seqs</span><span class='op'>(</span><span class='va'>...</span><span class='op'>)</span>

<span class='fu'>bins</span><span class='op'>(</span><span class='va'>...</span>, .group <span class='op'>=</span> <span class='fu'>vars</span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'>track</span><span class='op'>(</span>.track_id <span class='op'>=</span> <span class='fl'>1</span>, <span class='va'>...</span>, .track_type <span class='op'>=</span> <span class='cn'>NULL</span>, .ignore <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span>

<span class='fu'>pull_feats</span><span class='op'>(</span><span class='va'>.x</span>, .track_id <span class='op'>=</span> <span class='fl'>1</span>, <span class='va'>...</span>, .ignore <span class='op'>=</span> <span class='st'>"genes"</span>, .geneify <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='fu'>pull_genes</span><span class='op'>(</span>
  <span class='va'>.x</span>,
  <span class='va'>...</span>,
  .gene_types <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"CDS"</span>, <span class='st'>"mRNA"</span>, <span class='st'>"tRNA"</span>, <span class='st'>"tmRNA"</span>, <span class='st'>"ncRNA"</span>, <span class='st'>"rRNA"</span><span class='op'>)</span>
<span class='op'>)</span>

<span class='fu'>pull_links</span><span class='op'>(</span><span class='va'>.x</span>, .track_id <span class='op'>=</span> <span class='fl'>1</span>, <span class='va'>...</span>, .ignore <span class='op'>=</span> <span class='cn'>NULL</span>, .adjacent_only <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='fu'>pull_seqs</span><span class='op'>(</span><span class='va'>.x</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='fu'>pull_bins</span><span class='op'>(</span><span class='va'>.x</span>, <span class='va'>...</span>, .group <span class='op'>=</span> <span class='fu'>vars</span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># S3 method for gggenomes_layout</span>
<span class='fu'>pull_bins</span><span class='op'>(</span><span class='va'>.x</span>, <span class='va'>...</span>, .group <span class='op'>=</span> <span class='fu'>vars</span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='fu'>pull_track</span><span class='op'>(</span><span class='va'>.x</span>, .track_id <span class='op'>=</span> <span class='fl'>1</span>, <span class='va'>...</span>, .track_type <span class='op'>=</span> <span class='cn'>NULL</span>, .ignore <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>.track_id</th>
      <td><p>The track to pull out, either as a literal variable name or
as a positive/negative integer giving the position from the left/right.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Logical predicates passed on to <a href='https://dplyr.tidyverse.org/reference/filter.html'>dplyr::filter</a>. "seqs", "feats",
"links". Affects position-based selection.</p></td>
    </tr>
    <tr>
      <th>.ignore</th>
      <td><p>track names to ignore when selecting by position.</p></td>
    </tr>
    <tr>
      <th>.geneify</th>
      <td><p>add dummy type, introns and geom_id column to play nicely
with geoms supporting multi-level and spliced gene models.</p></td>
    </tr>
    <tr>
      <th>.gene_types</th>
      <td><p>return only feats of this type (<code>type %in% .gene_types</code>)</p></td>
    </tr>
    <tr>
      <th>.adjacent_only</th>
      <td><p>filter for links connecting direct neighbors
(<code>abs(y-yend)==1)</code>)</p></td>
    </tr>
    <tr>
      <th>.group</th>
      <td><p>what variables to use in grouping of bins from seqs in addition
to <code>y</code> and <code>bin_id</code>. Use this to get additional shared variables from the
seqs table into the bins table.</p></td>
    </tr>
    <tr>
      <th>.track_type</th>
      <td><p>restrict to these types of tracks - any combination of
"seqs", "feats", "links".</p></td>
    </tr>
    <tr>
      <th>.x</th>
      <td><p>A gggenomes or gggenomes_layout object.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    
<ul>
<li><p><code>feats</code>: by default pulls out the first feat track not named
"genes".</p></li>
<li><p><code>genes</code>: pulls out the first feat track (genes), filtering
for records with <code>type=="CDS"</code>, and adding a dummy <code>gene_id</code> column if missing
to play nice with multi-exon <code>geom</code>s.</p></li>
<li><p><code>links</code>: by default pulls out the first link track.</p></li>
<li><p><code>seqs</code>: pulls out the seqs track (there is only one).</p></li>
<li><p><code>bins</code>: pulls out a binwise summary table of the seqs data powering
<code>geom_bin_*()</code> calls. The bin table is not a real track, but recomputed
on-the-fly.</p></li>
<li><p><code>track</code>: pulls from all tracks in order seqs, feats, links.</p></li>
</ul>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='va'>gg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span><span class='va'>emale_genes</span>, <span class='va'>emale_seqs</span>, <span class='va'>emale_tirs</span>, <span class='va'>emale_ava</span><span class='op'>)</span>
<span class='va'>gg</span> <span class='op'>%&gt;%</span> <span class='fu'><a href='track_info.html'>track_info</a></span><span class='op'>(</span><span class='op'>)</span>            <span class='co'># info about track ids, positions and types</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 4 × 4</span>
#&gt; <span style='color: #949494;'># Groups:   type [3]</span>
#&gt;   id    type      i     n
#&gt;   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span>
#&gt; <span style='color: #BCBCBC;'>1</span> seqs  seqs      1     6
#&gt; <span style='color: #BCBCBC;'>2</span> genes feats     1   143
#&gt; <span style='color: #BCBCBC;'>3</span> feats feats     2    12
#&gt; <span style='color: #BCBCBC;'>4</span> links links     1    38</div><div class='input'>
<span class='co'># get first feat track that isn't "genes" (all equivalent)</span>
<span class='va'>gg</span> <span class='op'>%&gt;%</span> <span class='fu'>pull_feats</span><span class='op'>(</span><span class='op'>)</span>            <span class='co'># easiest</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 12 × 20</span>
#&gt;        y     x  xend bin_id      seq_id start   end file_id strand type  feat_id
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  
#&gt; <span style='color: #BCBCBC;'> 1</span>     2     0   473 BVI_008A    BVI_0…     1   473 emales… +      repe… BVI_00…
#&gt; <span style='color: #BCBCBC;'> 2</span>     2 <span style='text-decoration: underline;'>26</span>820 <span style='text-decoration: underline;'>26</span>347 BVI_008A    BVI_0… <span style='text-decoration: underline;'>26</span>348 <span style='text-decoration: underline;'>26</span>820 emales… -      repe… BVI_00…
#&gt; <span style='color: #BCBCBC;'> 3</span>     3     0   488 BVI_069     BVI_0…     1   488 emales… +      repe… BVI_06…
#&gt; <span style='color: #BCBCBC;'> 4</span>     3 <span style='text-decoration: underline;'>26</span>808 <span style='text-decoration: underline;'>26</span>320 BVI_069     BVI_0… <span style='text-decoration: underline;'>26</span>321 <span style='text-decoration: underline;'>26</span>808 emales… -      repe… BVI_06…
#&gt; <span style='color: #BCBCBC;'> 5</span>     4     0  <span style='text-decoration: underline;'>1</span>081 Cflag_017B  Cflag…     1  <span style='text-decoration: underline;'>1</span>081 emales… +      repe… Cflag_…
#&gt; <span style='color: #BCBCBC;'> 6</span>     4 <span style='text-decoration: underline;'>21</span>311 <span style='text-decoration: underline;'>20</span>230 Cflag_017B  Cflag… <span style='text-decoration: underline;'>20</span>231 <span style='text-decoration: underline;'>21</span>311 emales… -      repe… Cflag_…
#&gt; <span style='color: #BCBCBC;'> 7</span>     5     0   319 E4-10_086   E4-10…     1   319 emales… +      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'> 8</span>     5 <span style='text-decoration: underline;'>20</span>642 <span style='text-decoration: underline;'>20</span>323 E4-10_086   E4-10… <span style='text-decoration: underline;'>20</span>324 <span style='text-decoration: underline;'>20</span>642 emales… -      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'> 9</span>     1     0   458 E4-10_112   E4-10…     1   458 emales… +      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'>10</span>     1 <span style='text-decoration: underline;'>26</span>856 <span style='text-decoration: underline;'>26</span>398 E4-10_112   E4-10… <span style='text-decoration: underline;'>26</span>399 <span style='text-decoration: underline;'>26</span>856 emales… -      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'>11</span>     6     0   454 RCC970_016B RCC97…     1   454 emales… +      repe… RCC970…
#&gt; <span style='color: #BCBCBC;'>12</span>     6 <span style='text-decoration: underline;'>20</span>152 <span style='text-decoration: underline;'>19</span>698 RCC970_016B RCC97… <span style='text-decoration: underline;'>19</span>699 <span style='text-decoration: underline;'>20</span>152 emales… -      repe… RCC970…
#&gt; <span style='color: #949494;'># … with 9 more variables: introns &lt;list&gt;, parent_ids &lt;list&gt;, source &lt;chr&gt;,</span>
#&gt; <span style='color: #949494;'>#   score &lt;chr&gt;, phase &lt;int&gt;, name &lt;chr&gt;, width &lt;chr&gt;, geom_id &lt;chr&gt;,</span>
#&gt; <span style='color: #949494;'>#   .marginal &lt;lgl&gt;</span></div><div class='input'><span class='va'>gg</span> <span class='op'>%&gt;%</span> <span class='fu'>pull_feats</span><span class='op'>(</span><span class='va'>feats</span><span class='op'>)</span>       <span class='co'># by id</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 12 × 20</span>
#&gt;        y     x  xend bin_id      seq_id start   end file_id strand type  feat_id
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  
#&gt; <span style='color: #BCBCBC;'> 1</span>     2     0   473 BVI_008A    BVI_0…     1   473 emales… +      repe… BVI_00…
#&gt; <span style='color: #BCBCBC;'> 2</span>     2 <span style='text-decoration: underline;'>26</span>820 <span style='text-decoration: underline;'>26</span>347 BVI_008A    BVI_0… <span style='text-decoration: underline;'>26</span>348 <span style='text-decoration: underline;'>26</span>820 emales… -      repe… BVI_00…
#&gt; <span style='color: #BCBCBC;'> 3</span>     3     0   488 BVI_069     BVI_0…     1   488 emales… +      repe… BVI_06…
#&gt; <span style='color: #BCBCBC;'> 4</span>     3 <span style='text-decoration: underline;'>26</span>808 <span style='text-decoration: underline;'>26</span>320 BVI_069     BVI_0… <span style='text-decoration: underline;'>26</span>321 <span style='text-decoration: underline;'>26</span>808 emales… -      repe… BVI_06…
#&gt; <span style='color: #BCBCBC;'> 5</span>     4     0  <span style='text-decoration: underline;'>1</span>081 Cflag_017B  Cflag…     1  <span style='text-decoration: underline;'>1</span>081 emales… +      repe… Cflag_…
#&gt; <span style='color: #BCBCBC;'> 6</span>     4 <span style='text-decoration: underline;'>21</span>311 <span style='text-decoration: underline;'>20</span>230 Cflag_017B  Cflag… <span style='text-decoration: underline;'>20</span>231 <span style='text-decoration: underline;'>21</span>311 emales… -      repe… Cflag_…
#&gt; <span style='color: #BCBCBC;'> 7</span>     5     0   319 E4-10_086   E4-10…     1   319 emales… +      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'> 8</span>     5 <span style='text-decoration: underline;'>20</span>642 <span style='text-decoration: underline;'>20</span>323 E4-10_086   E4-10… <span style='text-decoration: underline;'>20</span>324 <span style='text-decoration: underline;'>20</span>642 emales… -      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'> 9</span>     1     0   458 E4-10_112   E4-10…     1   458 emales… +      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'>10</span>     1 <span style='text-decoration: underline;'>26</span>856 <span style='text-decoration: underline;'>26</span>398 E4-10_112   E4-10… <span style='text-decoration: underline;'>26</span>399 <span style='text-decoration: underline;'>26</span>856 emales… -      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'>11</span>     6     0   454 RCC970_016B RCC97…     1   454 emales… +      repe… RCC970…
#&gt; <span style='color: #BCBCBC;'>12</span>     6 <span style='text-decoration: underline;'>20</span>152 <span style='text-decoration: underline;'>19</span>698 RCC970_016B RCC97… <span style='text-decoration: underline;'>19</span>699 <span style='text-decoration: underline;'>20</span>152 emales… -      repe… RCC970…
#&gt; <span style='color: #949494;'># … with 9 more variables: introns &lt;list&gt;, parent_ids &lt;list&gt;, source &lt;chr&gt;,</span>
#&gt; <span style='color: #949494;'>#   score &lt;chr&gt;, phase &lt;int&gt;, name &lt;chr&gt;, width &lt;chr&gt;, geom_id &lt;chr&gt;,</span>
#&gt; <span style='color: #949494;'>#   .marginal &lt;lgl&gt;</span></div><div class='input'><span class='va'>gg</span> <span class='op'>%&gt;%</span> <span class='fu'>pull_feats</span><span class='op'>(</span><span class='fl'>1</span><span class='op'>)</span>           <span class='co'># by position</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 12 × 20</span>
#&gt;        y     x  xend bin_id      seq_id start   end file_id strand type  feat_id
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  
#&gt; <span style='color: #BCBCBC;'> 1</span>     2     0   473 BVI_008A    BVI_0…     1   473 emales… +      repe… BVI_00…
#&gt; <span style='color: #BCBCBC;'> 2</span>     2 <span style='text-decoration: underline;'>26</span>820 <span style='text-decoration: underline;'>26</span>347 BVI_008A    BVI_0… <span style='text-decoration: underline;'>26</span>348 <span style='text-decoration: underline;'>26</span>820 emales… -      repe… BVI_00…
#&gt; <span style='color: #BCBCBC;'> 3</span>     3     0   488 BVI_069     BVI_0…     1   488 emales… +      repe… BVI_06…
#&gt; <span style='color: #BCBCBC;'> 4</span>     3 <span style='text-decoration: underline;'>26</span>808 <span style='text-decoration: underline;'>26</span>320 BVI_069     BVI_0… <span style='text-decoration: underline;'>26</span>321 <span style='text-decoration: underline;'>26</span>808 emales… -      repe… BVI_06…
#&gt; <span style='color: #BCBCBC;'> 5</span>     4     0  <span style='text-decoration: underline;'>1</span>081 Cflag_017B  Cflag…     1  <span style='text-decoration: underline;'>1</span>081 emales… +      repe… Cflag_…
#&gt; <span style='color: #BCBCBC;'> 6</span>     4 <span style='text-decoration: underline;'>21</span>311 <span style='text-decoration: underline;'>20</span>230 Cflag_017B  Cflag… <span style='text-decoration: underline;'>20</span>231 <span style='text-decoration: underline;'>21</span>311 emales… -      repe… Cflag_…
#&gt; <span style='color: #BCBCBC;'> 7</span>     5     0   319 E4-10_086   E4-10…     1   319 emales… +      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'> 8</span>     5 <span style='text-decoration: underline;'>20</span>642 <span style='text-decoration: underline;'>20</span>323 E4-10_086   E4-10… <span style='text-decoration: underline;'>20</span>324 <span style='text-decoration: underline;'>20</span>642 emales… -      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'> 9</span>     1     0   458 E4-10_112   E4-10…     1   458 emales… +      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'>10</span>     1 <span style='text-decoration: underline;'>26</span>856 <span style='text-decoration: underline;'>26</span>398 E4-10_112   E4-10… <span style='text-decoration: underline;'>26</span>399 <span style='text-decoration: underline;'>26</span>856 emales… -      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'>11</span>     6     0   454 RCC970_016B RCC97…     1   454 emales… +      repe… RCC970…
#&gt; <span style='color: #BCBCBC;'>12</span>     6 <span style='text-decoration: underline;'>20</span>152 <span style='text-decoration: underline;'>19</span>698 RCC970_016B RCC97… <span style='text-decoration: underline;'>19</span>699 <span style='text-decoration: underline;'>20</span>152 emales… -      repe… RCC970…
#&gt; <span style='color: #949494;'># … with 9 more variables: introns &lt;list&gt;, parent_ids &lt;list&gt;, source &lt;chr&gt;,</span>
#&gt; <span style='color: #949494;'>#   score &lt;chr&gt;, phase &lt;int&gt;, name &lt;chr&gt;, width &lt;chr&gt;, geom_id &lt;chr&gt;,</span>
#&gt; <span style='color: #949494;'>#   .marginal &lt;lgl&gt;</span></div><div class='input'><span class='va'>gg</span> <span class='op'>%&gt;%</span> <span class='fu'>pull_feats</span><span class='op'>(</span><span class='fl'>2</span>, .ignore<span class='op'>=</span><span class='cn'>NULL</span><span class='op'>)</span>  <span class='co'># default .ignore="genes"</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 12 × 20</span>
#&gt;        y     x  xend bin_id      seq_id start   end file_id strand type  feat_id
#&gt;    <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  
#&gt; <span style='color: #BCBCBC;'> 1</span>     2     0   473 BVI_008A    BVI_0…     1   473 emales… +      repe… BVI_00…
#&gt; <span style='color: #BCBCBC;'> 2</span>     2 <span style='text-decoration: underline;'>26</span>820 <span style='text-decoration: underline;'>26</span>347 BVI_008A    BVI_0… <span style='text-decoration: underline;'>26</span>348 <span style='text-decoration: underline;'>26</span>820 emales… -      repe… BVI_00…
#&gt; <span style='color: #BCBCBC;'> 3</span>     3     0   488 BVI_069     BVI_0…     1   488 emales… +      repe… BVI_06…
#&gt; <span style='color: #BCBCBC;'> 4</span>     3 <span style='text-decoration: underline;'>26</span>808 <span style='text-decoration: underline;'>26</span>320 BVI_069     BVI_0… <span style='text-decoration: underline;'>26</span>321 <span style='text-decoration: underline;'>26</span>808 emales… -      repe… BVI_06…
#&gt; <span style='color: #BCBCBC;'> 5</span>     4     0  <span style='text-decoration: underline;'>1</span>081 Cflag_017B  Cflag…     1  <span style='text-decoration: underline;'>1</span>081 emales… +      repe… Cflag_…
#&gt; <span style='color: #BCBCBC;'> 6</span>     4 <span style='text-decoration: underline;'>21</span>311 <span style='text-decoration: underline;'>20</span>230 Cflag_017B  Cflag… <span style='text-decoration: underline;'>20</span>231 <span style='text-decoration: underline;'>21</span>311 emales… -      repe… Cflag_…
#&gt; <span style='color: #BCBCBC;'> 7</span>     5     0   319 E4-10_086   E4-10…     1   319 emales… +      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'> 8</span>     5 <span style='text-decoration: underline;'>20</span>642 <span style='text-decoration: underline;'>20</span>323 E4-10_086   E4-10… <span style='text-decoration: underline;'>20</span>324 <span style='text-decoration: underline;'>20</span>642 emales… -      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'> 9</span>     1     0   458 E4-10_112   E4-10…     1   458 emales… +      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'>10</span>     1 <span style='text-decoration: underline;'>26</span>856 <span style='text-decoration: underline;'>26</span>398 E4-10_112   E4-10… <span style='text-decoration: underline;'>26</span>399 <span style='text-decoration: underline;'>26</span>856 emales… -      repe… E4-10_…
#&gt; <span style='color: #BCBCBC;'>11</span>     6     0   454 RCC970_016B RCC97…     1   454 emales… +      repe… RCC970…
#&gt; <span style='color: #BCBCBC;'>12</span>     6 <span style='text-decoration: underline;'>20</span>152 <span style='text-decoration: underline;'>19</span>698 RCC970_016B RCC97… <span style='text-decoration: underline;'>19</span>699 <span style='text-decoration: underline;'>20</span>152 emales… -      repe… RCC970…
#&gt; <span style='color: #949494;'># … with 9 more variables: introns &lt;list&gt;, parent_ids &lt;list&gt;, source &lt;chr&gt;,</span>
#&gt; <span style='color: #949494;'>#   score &lt;chr&gt;, phase &lt;int&gt;, name &lt;chr&gt;, width &lt;chr&gt;, geom_id &lt;chr&gt;,</span>
#&gt; <span style='color: #949494;'>#   .marginal &lt;lgl&gt;</span></div><div class='input'>
<span class='co'># get "seqs" track (always track #1)</span>
<span class='va'>gg</span> <span class='op'>%&gt;%</span> <span class='fu'>pull_seqs</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span style='color: #949494;'># A tibble: 6 × 12</span>
#&gt; <span style='color: #949494;'># Groups:   bin_id [6]</span>
#&gt;       y     x  xend strand seq_id  bin_id  length bin_offset start   end file_id
#&gt;   <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;dbl&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>  
#&gt; <span style='color: #BCBCBC;'>1</span>     6     0 <span style='text-decoration: underline;'>20</span>152 +      RCC970… RCC970…  <span style='text-decoration: underline;'>20</span>152          0     1 <span style='text-decoration: underline;'>20</span>152 emales 
#&gt; <span style='color: #BCBCBC;'>2</span>     5     0 <span style='text-decoration: underline;'>20</span>642 +      E4-10_… E4-10_…  <span style='text-decoration: underline;'>20</span>642          0     1 <span style='text-decoration: underline;'>20</span>642 emales 
#&gt; <span style='color: #BCBCBC;'>3</span>     4     0 <span style='text-decoration: underline;'>21</span>311 +      Cflag_… Cflag_…  <span style='text-decoration: underline;'>21</span>311          0     1 <span style='text-decoration: underline;'>21</span>311 emales 
#&gt; <span style='color: #BCBCBC;'>4</span>     3     0 <span style='text-decoration: underline;'>26</span>808 +      BVI_069 BVI_069  <span style='text-decoration: underline;'>26</span>808          0     1 <span style='text-decoration: underline;'>26</span>808 emales 
#&gt; <span style='color: #BCBCBC;'>5</span>     2     0 <span style='text-decoration: underline;'>26</span>820 +      BVI_00… BVI_00…  <span style='text-decoration: underline;'>26</span>820          0     1 <span style='text-decoration: underline;'>26</span>820 emales 
#&gt; <span style='color: #BCBCBC;'>6</span>     1     0 <span style='text-decoration: underline;'>26</span>856 +      E4-10_… E4-10_…  <span style='text-decoration: underline;'>26</span>856          0     1 <span style='text-decoration: underline;'>26</span>856 emales 
#&gt; <span style='color: #949494;'># … with 1 more variable: seq_desc &lt;chr&gt;</span></div><div class='input'>
<span class='co'># plot integrated transposons and GC content for some viral genomes</span>
<span class='va'>gg</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gggenomes.html'>gggenomes</a></span><span class='op'>(</span>seqs<span class='op'>=</span><span class='va'>emale_seqs</span>, feats<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>emale_ngaros</span>, GC<span class='op'>=</span><span class='va'>emale_gc</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>gg</span> <span class='op'>+</span> <span class='fu'><a href='geom_seq.html'>geom_seq</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='geom_feat.html'>geom_feat</a></span><span class='op'>(</span>color<span class='op'>=</span><span class='st'>"skyblue"</span><span class='op'>)</span> <span class='op'>+</span> <span class='co'># defaults to data=feats()</span>
  <span class='fu'>geom_line</span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>y</span><span class='op'>+</span><span class='va'>score</span><span class='op'>-</span><span class='fl'>.6</span>, group<span class='op'>=</span><span class='va'>y</span><span class='op'>)</span>, data<span class='op'>=</span><span class='fu'>feats</span><span class='op'>(</span><span class='va'>GC</span><span class='op'>)</span>, color<span class='op'>=</span><span class='st'>"gray60"</span><span class='op'>)</span>
</div><div class='img'><img src='pull_track-1.png' alt='' width='960' height='480' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Thomas Hackl, Markus J. Ankenbrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


