<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read files in various formats into track tables — read_tracks • gggenomes</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  <script src="../extra.js"></script>

<meta property="og:title" content="Read files in various formats into track tables — read_tracks" />
<meta property="og:description" content="Convenience functions to read sequences, features or links from various
bioinformatics file formats, such as FASTA, GFF3, Genbank, BLAST tabular
output, etc. See def_formats() for full list. File formats and the
corresponding read-functions are automatically determined based on file
extensions. All these functions can read multiple files in the same format at
once, and combine them into a single table - useful, for example, to read a
folder of gff-files with each file containing genes of a different genome." />
<meta property="og:image" content="/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.8.2.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thackl/gggenomes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read files in various formats into track tables</h1>
    <small class="dont-index">Source: <a href='https://github.com/thackl/gggenomes/blob/master/R/read.R'><code>R/read.R</code></a>, <a href='https://github.com/thackl/gggenomes/blob/master/R/read_feats.R'><code>R/read_feats.R</code></a>, <a href='https://github.com/thackl/gggenomes/blob/master/R/read_seqs.R'><code>R/read_seqs.R</code></a></small>
    <div class="hidden name"><code>read_tracks.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Convenience functions to read sequences, features or links from various
bioinformatics file formats, such as FASTA, GFF3, Genbank, BLAST tabular
output, etc. See <code><a href='def_formats.html'>def_formats()</a></code> for full list. File formats and the
corresponding read-functions are automatically determined based on file
extensions. All these functions can read multiple files in the same format at
once, and combine them into a single table - useful, for example, to read a
folder of gff-files with each file containing genes of a different genome.</p>
    </div>

    <pre class="usage"><span class='fu'>read_feats</span><span class='op'>(</span><span class='va'>files</span>, .id <span class='op'>=</span> <span class='st'>"file_id"</span>, format <span class='op'>=</span> <span class='cn'>NULL</span>, parser <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='fu'>read_subfeats</span><span class='op'>(</span><span class='va'>files</span>, .id <span class='op'>=</span> <span class='st'>"file_id"</span>, format <span class='op'>=</span> <span class='cn'>NULL</span>, parser <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='fu'>read_links</span><span class='op'>(</span><span class='va'>files</span>, .id <span class='op'>=</span> <span class='st'>"file_id"</span>, format <span class='op'>=</span> <span class='cn'>NULL</span>, parser <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='fu'>read_sublinks</span><span class='op'>(</span><span class='va'>files</span>, .id <span class='op'>=</span> <span class='st'>"file_id"</span>, format <span class='op'>=</span> <span class='cn'>NULL</span>, parser <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='fu'>read_seqs</span><span class='op'>(</span>
  <span class='va'>files</span>,
  .id <span class='op'>=</span> <span class='st'>"file_id"</span>,
  format <span class='op'>=</span> <span class='cn'>NULL</span>,
  parser <span class='op'>=</span> <span class='cn'>NULL</span>,
  parse_desc <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>files</th>
      <td><p>files to reads. Should all be of same format. In many cases,
compressed files (<code>.gz</code>, <code>.bz2</code>, <code>.xz</code>, or <code>.zip</code>) are supported.
Similarly, automatic download of remote files starting with <code>http(s)://</code> or
<code>ftp(s)://</code> works in most cases.</p></td>
    </tr>
    <tr>
      <th>.id</th>
      <td><p>the column with the name of the file a record was read from.
Defaults to "file_id". Set to "bin_id" if every file represents a different
bin.</p></td>
    </tr>
    <tr>
      <th>format</th>
      <td><p>specify a format known to gggenomes, such as <code>gff3</code>, <code>gbk</code>, ...
to overwrite automatic determination based on the file extension (see
<code><a href='def_formats.html'>def_formats()</a></code> for full list).</p></td>
    </tr>
    <tr>
      <th>parser</th>
      <td><p>specify the name of an R function to overwrite automatic
determination based on format, e.g. <code>parser="read_tsv"</code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional arguments passed on to the format-specific read
function called down the line.</p></td>
    </tr>
    <tr>
      <th>parse_desc</th>
      <td><p>turn <code>key=some value</code> pairs from <code>seq_desc</code> into <code>key</code>-named
columns and remove them from <code>seq_desc</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A gggenomes-compatible sequence, feature or link tibble</p>
    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    
<ul>
<li><p><code>read_feats</code>: read files as features mapping onto
sequences.</p></li>
<li><p><code>read_subfeats</code>: read files as subfeatures mapping onto other features</p></li>
<li><p><code>read_links</code>: read files as links connecting sequences</p></li>
<li><p><code>read_sublinks</code>: read files as sublinks connecting features</p></li>
<li><p><code>read_seqs</code>: read sequence ID, description and length.</p></li>
</ul>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># read genes/features from a gff file</span>
<span class='fu'>read_feats</span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"eden-utr.gff"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'gff3' with `read_gff3()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: eden-utr [/home/runner/work/_temp/Library/gggenomes/extdata/eden-utr.gff]</span></div><div class='output co'>#&gt; <span class='message'>Harmonizing column names</span>
#&gt; <span class='message'>* ID -&gt; feat_id</span>
#&gt; <span class='message'>* Parent -&gt; parent_ids</span>
#&gt; <span class='message'>* Name -&gt; name</span></div><div class='output co'>#&gt; <span class='message'>Features read</span></div><div class='output co'>#&gt; <span class='message'><span style='color: #949494;'># A tibble: 8 x 3</span><span></span>
#&gt; <span class='message'>  source type                n</span>
#&gt; <span class='message'>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>           </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span></span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>1</span><span> </span><span style='color: #BB0000;'>NA</span><span>     cDNA_match          1</span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>2</span><span> </span><span style='color: #BB0000;'>NA</span><span>     CDS                 5</span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>3</span><span> </span><span style='color: #BB0000;'>NA</span><span>     exon                5</span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>4</span><span> </span><span style='color: #BB0000;'>NA</span><span>     five_prime_UTR      1</span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>5</span><span> </span><span style='color: #BB0000;'>NA</span><span>     gene                1</span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>6</span><span> </span><span style='color: #BB0000;'>NA</span><span>     mRNA                5</span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>7</span><span> </span><span style='color: #BB0000;'>NA</span><span>     TF_binding_site     1</span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>8</span><span> </span><span style='color: #BB0000;'>NA</span><span>     three_prime_UTR     1</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 20 x 15</span><span>
#&gt;    file_id  seq_id start   end strand type     feat_id introns parent_ids source
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;list&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;list&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> 
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>050  </span><span style='text-decoration: underline;'>9</span><span>000 +      cDNA_ma… match0… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>201  </span><span style='text-decoration: underline;'>7</span><span>600 +      CDS      cds000… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>201  </span><span style='text-decoration: underline;'>7</span><span>600 +      CDS      cds000… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>3</span><span>301  </span><span style='text-decoration: underline;'>7</span><span>600 +      CDS      cds000… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>3</span><span>391  </span><span style='text-decoration: underline;'>7</span><span>600 +      CDS      cds000… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>201  </span><span style='text-decoration: underline;'>7</span><span>600 +      CDS      cds000… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>300  </span><span style='text-decoration: underline;'>1</span><span>500 +      exon     exon00… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>050  </span><span style='text-decoration: underline;'>1</span><span>500 +      exon     exon00… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [2]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>3</span><span>000  </span><span style='text-decoration: underline;'>3</span><span>902 +      exon     exon00… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [2]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>5</span><span>000  </span><span style='text-decoration: underline;'>5</span><span>500 +      exon     exon00… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [3]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>7</span><span>000  </span><span style='text-decoration: underline;'>9</span><span>000 +      exon     exon00… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [3]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>12</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>050  </span><span style='text-decoration: underline;'>1</span><span>200 +      five_pr… f25     </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>13</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>000  </span><span style='text-decoration: underline;'>9</span><span>000 +      gene     gene00… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>14</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>050  </span><span style='text-decoration: underline;'>9</span><span>000 +      mRNA     mRNA00… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>15</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>050  </span><span style='text-decoration: underline;'>9</span><span>000 +      mRNA     mRNA00… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>16</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>300  </span><span style='text-decoration: underline;'>9</span><span>000 +      mRNA     mRNA00… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>17</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>300  </span><span style='text-decoration: underline;'>9</span><span>000 +      mRNA     mRNA00… </span><span style='color: #949494;'>&lt;dbl [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>18</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>050  </span><span style='text-decoration: underline;'>9</span><span>000 +      mRNA     mRNA00… </span><span style='color: #949494;'>&lt;int [</span><span>… </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>19</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>1</span><span>000  </span><span style='text-decoration: underline;'>1</span><span>012 +      TF_bind… tfbs00… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>20</span><span> eden-utr ctg123  </span><span style='text-decoration: underline;'>7</span><span>601  </span><span style='text-decoration: underline;'>9</span><span>000 +      three_p… f30     </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #949494;'># … with 5 more variables: score &lt;chr&gt;, phase &lt;int&gt;, name &lt;chr&gt;, Target &lt;chr&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   geom_id &lt;chr&gt;</span><span></div><div class='input'>

<span class='co'># read all gff files from a directory</span>
<span class='fu'>read_feats</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.files.html'>list.files</a></span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/"</span><span class='op'>)</span>, <span class='st'>"*.gff$"</span>, full.names<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'gff3' with `read_gff3()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales-ngaros [/home/runner/work/_temp/Library/gggenomes/extdata/emales//emales-ngaros.gff]</span></div><div class='output co'>#&gt; <span class='message'>Harmonizing column names</span>
#&gt; <span class='message'>* ID -&gt; feat_id</span>
#&gt; <span class='message'>* Parent -&gt; parent_ids</span>
#&gt; <span class='message'>* Name -&gt; name</span></div><div class='output co'>#&gt; <span class='message'>Features read</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #949494;'># A tibble: 1 x 3</span><span></span>
#&gt; <span class='message'>  source type              n</span>
#&gt; <span class='message'>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>         </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span></span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>1</span><span> MFG    repeat_region     3</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales-tirs [/home/runner/work/_temp/Library/gggenomes/extdata/emales//emales-tirs.gff]</span></div><div class='output co'>#&gt; <span class='message'>Harmonizing column names</span>
#&gt; <span class='message'>* ID -&gt; feat_id</span>
#&gt; <span class='message'>* Parent -&gt; parent_ids</span>
#&gt; <span class='message'>* Name -&gt; name</span></div><div class='output co'>#&gt; <span class='message'>Features read</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #949494;'># A tibble: 1 x 3</span><span></span>
#&gt; <span class='message'>  source type              n</span>
#&gt; <span class='message'>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>         </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span></span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>1</span><span> MFG    repeat_region    12</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales [/home/runner/work/_temp/Library/gggenomes/extdata/emales//emales.gff]</span></div><div class='output co'>#&gt; <span class='message'>Harmonizing column names</span>
#&gt; <span class='message'>* ID -&gt; feat_id</span>
#&gt; <span class='message'>* Parent -&gt; parent_ids</span>
#&gt; <span class='message'>* Name -&gt; name</span></div><div class='output co'>#&gt; <span class='message'>Features read</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #949494;'># A tibble: 1 x 3</span><span></span>
#&gt; <span class='message'>  source type      n</span>
#&gt; <span class='message'>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span></span>
#&gt; <span class='message'></span><span style='color: #BCBCBC;'>1</span><span> MFG    CDS     143</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 158 x 17</span><span>
#&gt;    file_id  seq_id  start   end strand type   feat_id  introns parent_ids source
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;list&gt;</span><span>  </span><span style='color: #949494;font-style: italic;'>&lt;list&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> 
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> emales-… BVI_00… </span><span style='text-decoration: underline;'>11</span><span>043 </span><span style='text-decoration: underline;'>17</span><span>065 </span><span style='color: #BB0000;'>NA</span><span>     repea… nBVI_008 </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> emales-… BVI_069  </span><span style='text-decoration: underline;'>7</span><span>894 </span><span style='text-decoration: underline;'>14</span><span>280 </span><span style='color: #BB0000;'>NA</span><span>     repea… nBVI_069 </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> emales-… E4-10_…  </span><span style='text-decoration: underline;'>2</span><span>854  </span><span style='text-decoration: underline;'>9</span><span>283 </span><span style='color: #BB0000;'>NA</span><span>     repea… nE4-10_… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> emales-… BVI_00…     1   473 +      repea… BVI_008… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> emales-… BVI_00… </span><span style='text-decoration: underline;'>26</span><span>348 </span><span style='text-decoration: underline;'>26</span><span>820 -      repea… BVI_008… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> emales-… BVI_069     1   488 +      repea… BVI_069… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> emales-… BVI_069 </span><span style='text-decoration: underline;'>26</span><span>321 </span><span style='text-decoration: underline;'>26</span><span>808 -      repea… BVI_069… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> emales-… Cflag_…     1  </span><span style='text-decoration: underline;'>1</span><span>081 +      repea… Cflag_0… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> emales-… Cflag_… </span><span style='text-decoration: underline;'>20</span><span>231 </span><span style='text-decoration: underline;'>21</span><span>311 -      repea… Cflag_0… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> emales-… E4-10_…     1   319 +      repea… E4-10_0… </span><span style='color: #949494;'>&lt;NULL&gt;</span><span>  </span><span style='color: #949494;'>&lt;chr [1]&gt;</span><span>  MFG   
#&gt; </span><span style='color: #949494;'># … with 148 more rows, and 7 more variables: score &lt;chr&gt;, phase &lt;int&gt;,</span><span>
#&gt; </span><span style='color: #949494;'>#   name &lt;chr&gt;, geom_id &lt;chr&gt;, width &lt;chr&gt;, gc_content &lt;chr&gt;, Note &lt;chr&gt;</span><span></div><div class='input'>

<span class='co'># read remote files</span>
<span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='va'>gbk_phages</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
  PSSP7 <span class='op'>=</span> <span class='st'>"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/858/745/GCF_000858745.1_ViralProj15134/GCF_000858745.1_ViralProj15134_genomic.gff.gz"</span>,
  PSSP3 <span class='op'>=</span> <span class='st'>"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/904/555/GCF_000904555.1_ViralProj195517/GCF_000904555.1_ViralProj195517_genomic.gff.gz"</span><span class='op'>)</span>
<span class='fu'>read_feats</span><span class='op'>(</span><span class='va'>gbk_phages</span><span class='op'>)</span>
<span class='op'>}</span>


<span class='co'># read sequences from a fasta file.</span>
<span class='fu'>read_seqs</span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.fna"</span><span class='op'>)</span>, parse_desc<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'fasta' with `read_seq_len()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.fna]</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 6 x 4</span><span>
#&gt;   file_id seq_id      seq_desc                                            length
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                                                </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> emales  BVI_008A    emale_type=EMALE01 is_typespecies=FALSE has_tir=TR…  </span><span style='text-decoration: underline;'>26</span><span>820
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> emales  BVI_069     emale_type=EMALE01 is_typespecies=FALSE has_tir=TR…  </span><span style='text-decoration: underline;'>26</span><span>808
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> emales  Cflag_017B  emale_type=EMALE01 is_typespecies=TRUE has_tir=TRUE  </span><span style='text-decoration: underline;'>21</span><span>311
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> emales  E4-10_086   emale_type=EMALE01 is_typespecies=FALSE has_tir=TR…  </span><span style='text-decoration: underline;'>20</span><span>642
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> emales  E4-10_112   emale_type=EMALE01 is_typespecies=FALSE has_tir=TR…  </span><span style='text-decoration: underline;'>26</span><span>856
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> emales  RCC970_016B emale_type=EMALE01 is_typespecies=FALSE has_tir=TR…  </span><span style='text-decoration: underline;'>20</span><span>152</div><div class='input'>
<span class='co'># read sequence info from a fasta file with `parse_desc=TRUE` (default). `key=value`</span>
<span class='co'># pairs are removed from `seq_desc` and parsed into columns with `key` as name</span>
<span class='fu'>read_seqs</span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.fna"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'fasta' with `read_seq_len()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.fna]</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 6 x 7</span><span>
#&gt;   file_id seq_id      seq_desc length emale_type is_typespecies has_tir
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>  
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> emales  BVI_008A    </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>820 EMALE01    FALSE          TRUE   
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> emales  BVI_069     </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>808 EMALE01    FALSE          TRUE   
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> emales  Cflag_017B  </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>21</span><span>311 EMALE01    TRUE           TRUE   
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> emales  E4-10_086   </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>20</span><span>642 EMALE01    FALSE          TRUE   
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> emales  E4-10_112   </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>856 EMALE01    FALSE          TRUE   
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> emales  RCC970_016B </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>20</span><span>152 EMALE01    FALSE          TRUE   </div><div class='input'>
<span class='co'># read sequence info from samtools/seqkit style index</span>
<span class='fu'>read_seqs</span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.fna.seqkit.fai"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'fai' with `read_fai()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales.fna.seqkit [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.fna.seqkit.fai]</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 6 x 7</span><span>
#&gt;   file_id           seq_id     seq_desc length emale_type is_typespecies has_tir
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>  
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> emales.fna.seqkit BVI_008A   </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>820 EMALE01    FALSE          TRUE   
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> emales.fna.seqkit BVI_069    </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>808 EMALE01    FALSE          TRUE   
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> emales.fna.seqkit Cflag_017B </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>21</span><span>311 EMALE01    TRUE           TRUE   
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> emales.fna.seqkit E4-10_086  </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>20</span><span>642 EMALE01    FALSE          TRUE   
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> emales.fna.seqkit E4-10_112  </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>856 EMALE01    FALSE          TRUE   
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> emales.fna.seqkit RCC970_01… </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>20</span><span>152 EMALE01    FALSE          TRUE   </div><div class='input'>
<span class='co'># read sequence info from multiple gff file</span>
<span class='fu'>read_seqs</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.gff"</span><span class='op'>)</span>, <span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales-tirs.gff"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading 'gff3' with `read_seq_len()`:</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.gff]</span></div><div class='output co'>#&gt; <span class='message'>* file_id: emales-tirs [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales-tirs.gff]</span></div><div class='output co'>#&gt; </span><span style='color: #949494;'># A tibble: 12 x 4</span><span>
#&gt;    file_id     seq_id      seq_desc length
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> emales      BVI_008A    </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>820
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> emales      BVI_069     </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>808
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> emales      Cflag_017B  </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>21</span><span>311
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> emales      E4-10_086   </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>20</span><span>642
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> emales      E4-10_112   </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>856
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> emales      RCC970_016B </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>20</span><span>152
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> emales-tirs BVI_008A    </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>820
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> emales-tirs BVI_069     </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>808
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> emales-tirs Cflag_017B  </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>21</span><span>311
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> emales-tirs E4-10_086   </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>20</span><span>642
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> emales-tirs E4-10_112   </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>26</span><span>856
#&gt; </span><span style='color: #BCBCBC;'>12</span><span> emales-tirs RCC970_016B </span><span style='color: #BB0000;'>NA</span><span>        </span><span style='text-decoration: underline;'>20</span><span>152</div><div class='input'>
</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Thomas Hackl, Markus J. Ankenbrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


