<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • gggenomes</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<script src="../extra.js"></script><meta property="og:title" content="">
<meta property="og:description" content="gggenomes">
<meta property="og:image" content="/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.2.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gggenomes.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes (old version)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/thackl/gggenomes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="gggenomes_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/thackl/gggenomes/blob/master/vignettes/gggenomes.Rmd"><code>vignettes/gggenomes.Rmd</code></a></small>
      <div class="hidden name"><code>gggenomes.Rmd</code></div>

    </div>

    
    
<p>q— title: “gggenomes - a multi-track ggplot2 extension for comparative genomics” author: “Thomas Hackl” date: “2/23/2021” output: pdf_document —</p>
<p>gggenomes is an extension to ggplot2 tailored to the visualization of genomics data. It provides the same flexible approach for building up plots layer by layer, but supports the simultaneous use of multiple datasets within one plot, thereby intuitively mirroring the relational nature of most ’omics data products.</p>
<p><strong>A ggplot2 is build around a single, tidy dataset</strong> with arbitrary variables and observations. x/y-plot coordinates for things to draw are computed on-the-fly using aesthetic mappings that link variables to coordinates. This setup is highly flexible and makes ggplot extremely powerful.</p>
<p><strong>A gggenomes plot is build around multiple data tracks</strong> that each contain a single dataset. These datasets are linked via pre-computed global layout that determines x/y-plot coordinates prior to the actual plot construction. The reason for this more rigid setup is that genomics data are relational and there is just no good way to represent all the necessary information in a single tidy table.</p>
<p>gggenomes knows three types of tracks:</p>
<ol style="list-style-type: decimal">
<li>
<strong>seqs</strong>: sequences such as contigs or chromosomes</li>
<li>
<strong>feats</strong>: annotations of locations on sequences, such genes, SNPs, …</li>
<li>
<strong>links</strong>: annotations that connect two locations between two different sequences</li>
</ol>
<p>There are also <strong><em>genes</em></strong> tracks, those, however are just glorified feature tracks, that sometimes get special treatment for easier usage. Internally, genes are features.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thackl/gggenomes">gggenomes</a></span><span class="op">)</span>

<span class="co"># a minimal seq track</span>
<span class="va">s0</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>
  seq_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"b"</span><span class="op">)</span>,
  length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">600</span>, <span class="fl">550</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># a minimal gene track</span>
<span class="va">g0</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>
  seq_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"a"</span>, <span class="st">"b"</span><span class="op">)</span>,
  start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">350</span>, <span class="fl">80</span><span class="op">)</span>,
  end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">250</span>, <span class="fl">500</span>, <span class="fl">450</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># a simple link track</span>
<span class="va">l0</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>
  seq_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"a"</span><span class="op">)</span>,
  start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">400</span><span class="op">)</span>,
  end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">250</span>, <span class="fl">480</span><span class="op">)</span>,
  seq_id2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"b"</span>, <span class="st">"b"</span><span class="op">)</span>,
  start2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">80</span>, <span class="fl">350</span><span class="op">)</span>,
  end2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">300</span>, <span class="fl">430</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gggenomes.html">gggenomes</a></span><span class="op">(</span>genes<span class="op">=</span><span class="va">g0</span>, seqs<span class="op">=</span><span class="va">s0</span>, links<span class="op">=</span><span class="va">l0</span><span class="op">)</span>
<span class="va">p</span> <span class="op">+</span> 
  <span class="fu"><a href="../reference/geom_seq.html">geom_seq</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>         <span class="co"># draw contig/chromosome lines</span>
  <span class="fu"><a href="../reference/geom_seq_label.html">geom_seq_label</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>   <span class="co"># label each sequence </span>
  <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>        <span class="co"># draw genes as arrow</span>
  <span class="fu"><a href="../reference/geom_link.html">geom_link</a></span><span class="op">(</span><span class="op">)</span>          <span class="co"># draw some connections between syntenic regions</span></code></pre></div>
<p><img src="gggenomes_files/figure-html/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;"></p>
<div id="geoms-and-tracks" class="section level1">
<h1 class="hasAnchor">
<a href="#geoms-and-tracks" class="anchor"></a>Geoms and tracks</h1>
<p>In a regular ggplot2, geoms read from one central dataframe. For gggenomes, there are several tracks to read data from. To specify which tracks to use in which geoms, gggenomes provides accessory functions that work inside geom calls when provided to the <code>data</code> argument (or outside if prefixed with <code>pull_</code>). The names of these functions mirror the track types they work on and should be quite intuitive:</p>
<ul>
<li>
<code>geom_seq()  : data=seqs()</code> for the seq track</li>
<li>
<code>geom_feat() : data=feats()</code> for the first feat track not named “genes”</li>
<li>
<code>geom_link() : data=links()</code> for the first link track</li>
<li>
<code>geom_gene() : data=genes()</code> for the first feat track with some extras for geneish features (see <a href="#inside-gggenomes-tracks">Inside gggenomes tracks</a> for details)</li>
</ul>
<p>gggenomes supports an arbitrary number of feature and link tracks within the same plot. Extra tracks can be added by either providing a list of tracks to <code><a href="../reference/gggenomes.html">gggenomes()</a></code>, or by using the dedicated <code><a href="../reference/add_feats.html">add_feats()</a></code> / <code><a href="../reference/add_feats.html">add_links()</a></code> functions. The extra tracks can then be used via the aforementioned accessory function inside geom calls using names or positional arguments. Like most gggenomes functions the track accessory functions use tidyverse-style non-standard evaluation, which means you can refer to tracks either by unquoted names or by numeric position.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Let's use some of the bundled example data here</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"gggenomes"</span><span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gggenomes.html">gggenomes</a></span><span class="op">(</span>
  genes<span class="op">=</span><span class="va">emale_genes</span>,  <span class="co"># a gene track, added as first feat track</span>
  seqs<span class="op">=</span><span class="va">emale_seqs</span>,    <span class="co"># a seq track</span>
  feats<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">emale_tirs</span>, <span class="va">emale_ngaros</span><span class="op">)</span>,  <span class="co"># multiple feat tracks</span>
  links<span class="op">=</span><span class="va">emale_ava</span>     <span class="co"># a link track</span>
<span class="op">)</span>

<span class="co"># inspect the tracks of the plot</span>
<span class="va">p</span> <span class="op">%&gt;%</span> <span class="va">track_info</span></code></pre></div>
<pre><code>## # A tibble: 5 × 4
## # Groups:   type [3]
##   id           type      i     n
##   &lt;chr&gt;        &lt;chr&gt; &lt;int&gt; &lt;int&gt;
## 1 seqs         seqs      1     6
## 2 genes        feats     1   143
## 3 emale_tirs   feats     2    12
## 4 emale_ngaros feats     3     3
## 5 links        links     1    38</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot all tracks</span>
<span class="va">p</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_link.html">geom_link</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>  <span class="co"># the first link track</span>
  <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>  <span class="co"># the first feat track filtered for geneish feats: CDS, mRNA, ..</span>
  <span class="fu"><a href="../reference/geom_feat.html">geom_feat</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>  <span class="co"># the first feat track not named "genes", here emale_tirs</span>
  <span class="co"># use an extra feat track by name</span>
  <span class="fu"><a href="../reference/geom_feat.html">geom_feat</a></span><span class="op">(</span>data<span class="op">=</span><span class="fu"><a href="../reference/feats.html">feats</a></span><span class="op">(</span><span class="va">emale_ngaros</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"plum3"</span><span class="op">)</span></code></pre></div>
<p><img src="gggenomes_files/figure-html/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="one-layout-to-rule-them-all" class="section level1">
<h1 class="hasAnchor">
<a href="#one-layout-to-rule-them-all" class="anchor"></a>One layout to rule them all</h1>
<p>gggenomes multi-track setup has both pros and cons. For example, communicating information between tracks during plot construction is not supported by ggplot2, because - well - it’s been designed to work with a single table. gggenomes works around that by <strong>pre-computing a layout</strong>, and adding coordinates (<code>y,x,xend</code>) to each dataframe prior to the actual plot construction. This has some implications for the usage of gggenomes:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Dataframes for tracks have required variables</strong>, e.g. <code>seq_id,start,end</code> for features. These predefined variables are used during import to compute x/y coordinates (see <a href="#inside-gggenomes-tracks">Inside gggenomes tracks</a> for more details).</li>
<li>
<strong>gggenomes geoms can often be used without explicit <code><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes()</a></code> mappings</strong> - a rather convenient upside of the multi-track setup. This works because we always know the names of the plot variables ahead of time: they originate from the pre-computed layout, and we can use that information to set sensible default aesthetic mappings for most cases.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># inspect seqs track with layout vars - note y,x,xend</span>
<span class="va">p</span> <span class="op">%&gt;%</span> <span class="va">pull_seqs</span></code></pre></div>
<pre><code>## # A tibble: 6 × 12
## # Groups:   bin_id [6]
##       y     x  xend strand seq_id  bin_id  length bin_offset start   end file_id
##   &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;    &lt;int&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;int&gt; &lt;chr&gt;  
## 1     6     0 20152 +      RCC970… RCC970…  20152          0     1 20152 emales 
## 2     5     0 20642 +      E4-10_… E4-10_…  20642          0     1 20642 emales 
## 3     4     0 21311 +      Cflag_… Cflag_…  21311          0     1 21311 emales 
## 4     3     0 26808 +      BVI_069 BVI_069  26808          0     1 26808 emales 
## 5     2     0 26820 +      BVI_00… BVI_00…  26820          0     1 26820 emales 
## 6     1     0 26856 +      E4-10_… E4-10_…  26856          0     1 26856 emales 
## # … with 1 more variable: seq_desc &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># inspect genes track with layout vars - note y,x,xend, but also other</span>
<span class="co"># columns such as strand, feat_id or type, that are added automatically</span>
<span class="va">p</span> <span class="op">%&gt;%</span> <span class="va">pull_genes</span></code></pre></div>
<pre><code>## # A tibble: 143 × 22
##        y     x  xend bin_id   seq_id   start   end file_id strand type  feat_id 
##    &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;int&gt; &lt;int&gt; &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;   
##  1     2   567   822 BVI_008A BVI_008A   568   822 emales  +      CDS   BVI_008…
##  2     2  2668  1039 BVI_008A BVI_008A  1040  2668 emales  -      CDS   BVI_008…
##  3     2  3028  2767 BVI_008A BVI_008A  2768  3028 emales  -      CDS   BVI_008…
##  4     2  4776  3138 BVI_008A BVI_008A  3139  4776 emales  -      CDS   BVI_008…
##  5     2  5370  4809 BVI_008A BVI_008A  4810  5370 emales  -      CDS   BVI_008…
##  6     2  6065  5396 BVI_008A BVI_008A  5397  6065 emales  -      CDS   BVI_008…
##  7     2  6683  6074 BVI_008A BVI_008A  6075  6683 emales  -      CDS   BVI_008…
##  8     2  6867  6540 BVI_008A BVI_008A  6541  6867 emales  -      CDS   BVI_008…
##  9     2  6954  7788 BVI_008A BVI_008A  6955  7788 emales  +      CDS   BVI_008…
## 10     2  7997  9026 BVI_008A BVI_008A  7998  9026 emales  +      CDS   BVI_008…
## # … with 133 more rows, and 11 more variables: introns &lt;list&gt;,
## #   parent_ids &lt;list&gt;, source &lt;chr&gt;, score &lt;chr&gt;, phase &lt;int&gt;, width &lt;chr&gt;,
## #   gc_content &lt;chr&gt;, name &lt;chr&gt;, Note &lt;chr&gt;, geom_id &lt;chr&gt;, .marginal &lt;lgl&gt;</code></pre>
</div>
<div id="plotting-from-scratch" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-from-scratch" class="anchor"></a>Plotting from scratch</h1>
<p>The minimum requirement to create a gggenomes plot is a single track. It can be of any of the three types: seqs, feats or links.</p>
<p>If there is no a sequence track, a sequence track is inferred from the data simply by assuming that each sequence that a feature or link maps onto must exist, and that it must be long enough to accommodate all its associated features and links (note the corresponding log message below).</p>
<p>Note also, that unless sequence information is explicitly provided, gggenomes will zoom in on only the regions that are covered with features. That may appear odd at first, but it is a) consistent in the sense that for inferred sequence we only know the start (<code>x=1</code>) anyway, and have to guess the end from the last feature, and b) it is very convenient if we deal with real data: It usually makes no sense to plot all 5000 genes of a bacterial genome. In most cases we will want to focus on specific regions.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># some genes</span>
<span class="va">g0</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>
  seq_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"a"</span>, <span class="st">"b"</span><span class="op">)</span>,
  start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">350</span>, <span class="fl">80</span><span class="op">)</span>,
  end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">250</span>, <span class="fl">500</span>, <span class="fl">450</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gggenomes.html">gggenomes</a></span><span class="op">(</span><span class="va">g0</span><span class="op">)</span> </code></pre></div>
<pre><code>## No seqs provided, inferring seqs from feats</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_seq.html">geom_seq</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>         <span class="co"># draw contig/chromosome lines</span>
  <span class="fu"><a href="../reference/geom_seq_label.html">geom_seq_label</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>   <span class="co"># label each sequence </span>
  <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span>          <span class="co"># draw genes as arrow</span></code></pre></div>
<p><img src="gggenomes_files/figure-html/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="plotting-from-files" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-from-files" class="anchor"></a>Plotting from files</h1>
<p>Plotting from scratch is fun, but likely not the typical use-case. Usually you will have files with data that you want to import. gggenomes provides a bunch of <code>read_*</code> functions that help you do that for several common bioinformatics file formats. Moreover, gggenomes come with a set of higher-level convenience read functions that perfectly integrate with its track system.</p>
<p>What I mean by that is that instead of reading a certain file format explicitly, such as <code><a href="../reference/read_gff3.html">read_gff3("foo.gff")</a></code>, you can invoke the generic <code><a href="../reference/read_tracks.html">read_feats("foo.gff")</a></code> function to read features from various file formats. File formats are automatically recognized, so this, for example, also works <code><a href="../reference/read_tracks.html">read_feats("foo.bed")</a></code>. And you can read different information from the same file: <code><a href="../reference/read_tracks.html">read_seqs("foo.gff")</a></code> will retrieve sequence information from the gff3 file instead of gene/feature coordinates.</p>
<p>That said, the only real requirement for any data you want to plot is that it is in a somewhat tidy table, and that it has some of the required variables (see below). Even if you have data that is not directly supported by gggenomes read functions, it should be very easy to convert your data into a table that works with gggenomes.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># note: ex() is just a helper to get stable paths to gggenomes example data</span>
<span class="va">s0</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_tracks.html">read_seqs</a></span><span class="op">(</span><span class="fu"><a href="../reference/ex.html">ex</a></span><span class="op">(</span><span class="st">"emales/emales.fna"</span><span class="op">)</span><span class="op">)</span>
<span class="va">g0</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_tracks.html">read_feats</a></span><span class="op">(</span><span class="fu"><a href="../reference/ex.html">ex</a></span><span class="op">(</span><span class="st">"emales/emales.gff"</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="../reference/gggenomes.html">gggenomes</a></span><span class="op">(</span><span class="va">g0</span>, <span class="va">s0</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_seq.html">geom_seq</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="gggenomes_files/figure-html/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># for lazy people</span>
<span class="fu"><a href="../reference/gggenomes.html">gggenomes</a></span><span class="op">(</span><span class="fu"><a href="../reference/ex.html">ex</a></span><span class="op">(</span><span class="st">"emales/emales.gff"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<p><img src="gggenomes_files/figure-html/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># and really fancy: multiple remote files, all at once</span>
<span class="va">gbk_phages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  PSSP7 <span class="op">=</span> <span class="st">"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/858/745/GCF_000858745.1_ViralProj15134/GCF_000858745.1_ViralProj15134_genomic.gff.gz"</span>,
  PSSP3 <span class="op">=</span> <span class="st">"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/904/555/GCF_000904555.1_ViralProj195517/GCF_000904555.1_ViralProj195517_genomic.gff.gz"</span><span class="op">)</span>
<span class="fu"><a href="../reference/gggenomes.html">gggenomes</a></span><span class="op">(</span><span class="va">gbk_phages</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_seq_label.html">geom_seq_label</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="gggenomes_files/figure-html/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="inside-gggenomes-tracks" class="section level1">
<h1 class="hasAnchor">
<a href="#inside-gggenomes-tracks" class="anchor"></a>Inside gggenomes tracks</h1>
<p>As mentioned before, tracks have required columns and optional columns, which can have additional meaning in the context of the plot.</p>
<p><strong>seqs</strong>: sequences such as contigs or chromosomes</p>
<ul>
<li>required: <code>seq_id,length</code>
</li>
<li>recognized: <code>bin_id,start,end,strand</code>
</li>
</ul>
<p><strong>feats</strong>: such as genes, SNPs, repeats, …</p>
<ul>
<li>required: <code>seq_id,start,end</code>
</li>
<li>recognized: <code>strand,bin_id,feat_id,introns</code>
</li>
</ul>
<p><strong>links</strong>: connections between two locations from different sequences</p>
<ul>
<li>required: <code>seq_id,seq_id2</code>
</li>
<li>recognized: <code>start,end,bin_id,start2,end2,bin_id2,strand</code>
</li>
</ul>
<div id="the-seq-track---there-can-be-only-one" class="section level2">
<h2 class="hasAnchor">
<a href="#the-seq-track---there-can-be-only-one" class="anchor"></a>The seq track - there can be only one</h2>
<p><strong><code>seq_id,length</code></strong> are the required variables and should be quite self-explanatory. It’s highly recommended (and might become compulsory in the future) for <code>seq_id</code>s to be unique within the entire dataset.</p>
<p>The optional column for seqs are:</p>
<p><strong><code>bin_id</code></strong> to denote <strong>bins</strong> representing genomes or assemblies with multiple chromosomes or contigs. Sequences from the same bin are drawn next to each other in the default layout, as opposed to sequence from different bins which will occupy different rows. If omitted, the default is to set <code>bin_id=seq_id</code>, which means, every sequence gets its own row.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># seq track: one entry per sequence</span>
<span class="va">s0</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>
  bin_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>,
  seq_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a1"</span>,<span class="st">"a2"</span>,<span class="st">"b1"</span><span class="op">)</span>,
  length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2e5</span>, <span class="fl">3e5</span>, <span class="fl">5e5</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/gggenomes.html">gggenomes</a></span><span class="op">(</span>seqs<span class="op">=</span><span class="va">s0</span><span class="op">)</span> 
<span class="va">p</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_seq.html">geom_seq</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>         <span class="co"># draw contig/chromosome lines</span>
  <span class="fu"><a href="../reference/geom_seq_label.html">geom_seq_label</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>   <span class="co"># label each sequence </span>
  <span class="fu"><a href="../reference/geom_bin_label.html">geom_bin_label</a></span><span class="op">(</span><span class="op">)</span>  <span class="co"># label each bin</span></code></pre></div>
<p><img src="gggenomes_files/figure-html/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;"></p>
<p><strong><code>start,end,strand</code></strong> in the context of a sequence denote a region or <strong><em>locus</em></strong> on that sequence. If provided, only the part of the sequence within the locus boundaries will be shown in the plot. This also will remove any features from the plot that do not fall within the locus. It’s likely rare to one wants to specify these information directly, but they are important as they power manipulation functions such as <code><a href="../reference/focus.html">focus()</a></code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># zoom in on a longer sequence - note the scale on the x-axis</span>
<span class="va">s0</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>
  seq_id <span class="op">=</span> <span class="st">"a1"</span>,
  length <span class="op">=</span> <span class="fl">10000</span>,
  start <span class="op">=</span> <span class="fl">1000</span>,
  end <span class="op">=</span> <span class="fl">3000</span>
<span class="op">)</span>
<span class="fu"><a href="../reference/gggenomes.html">gggenomes</a></span><span class="op">(</span>seqs<span class="op">=</span><span class="va">s0</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_seq.html">geom_seq</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_seq_label.html">geom_seq_label</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="gggenomes_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="feature-tracks---what-maps-onto-sequences" class="section level2">
<h2 class="hasAnchor">
<a href="#feature-tracks---what-maps-onto-sequences" class="anchor"></a>Feature tracks - what maps onto sequences</h2>
<p>Again, the required variables <code>seq_id,start,end</code> should be self-explainatory. Note though, that features with <code>seq_id</code>s not present in the seq track will be silently ignored.</p>
<p>Also note that gggenomes uses <strong>1-based, inclusive ranges</strong> to denote feature start/end coordinates. It’s the same system as used by GFF, GenBank, BLAST or SAM, for example. In contrast, BED and BAM, for example, use 0-based, exclusive ranges. Something to keep in mind when importing data from those formats.</p>
<p>The optional columns for features are:</p>
<p><strong><code>strand</code></strong> to denote the orientation of a feature relative to the sequences it maps to. The default way of specifying the strand is with a character vector with <code>"+/-/."</code> coding for <em>forward</em>, <em>reverse</em> and <em>undetermined</em>. Logical (<code>TRUE/FALSE/NA</code>) or numeric vectors (<code>1/-1/0</code>) are supported as well.</p>
<p><strong><code>bin_id</code></strong> works as for seqs. The assignment of <code>bin_id</code>s to features is usually only useful, if no sequences are provided. Otherwise, bin information is inherited from the sequences.</p>
<p><strong><code>feat_id</code></strong> is a unique ID for each feature. These IDs are useful if we want to add data to a plot that pertains to features and not sequences. For example, we can add the results of a protein blast search to the genes the proteins derived from if by matching blast results and feature annotations using the <code>feat_id</code>. If not provided, they will be automatically generated.</p>
<p>Special variables for complex gene models:</p>
<p><strong><code>introns</code></strong> is a special column that can contain a list of numeric vectors that denote intron start and end coordinates within a given feature. The support of introns is something I just very recently added. It’s still experimental and some of the handling might change in future versions.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># some genes</span>
<span class="va">g0</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>
  seq_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span><span class="op">)</span>,
  start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">800</span><span class="op">)</span>,
  end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">1200</span><span class="op">)</span>,
  <span class="co"># NOTE: introns need to be a list-column!</span>
  introns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">200</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>,<span class="fl">200</span>,<span class="fl">250</span>,<span class="fl">300</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="../reference/gggenomes.html">gggenomes</a></span><span class="op">(</span><span class="va">g0</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_seq.html">geom_seq</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>         <span class="co"># draw contig/chromosome lines</span>
  <span class="fu"><a href="../reference/geom_seq_label.html">geom_seq_label</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>   <span class="co"># label each sequence </span>
  <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span><span class="op">)</span>          <span class="co"># draw genes as arrow</span></code></pre></div>
<pre><code>## No seqs provided, inferring seqs from feats</code></pre>
<p><img src="gggenomes_files/figure-html/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div id="link-tracks---show-me-the-synteny" class="section level2">
<h2 class="hasAnchor">
<a href="#link-tracks---show-me-the-synteny" class="anchor"></a>Link tracks - show me the synteny</h2>
<p>Link tracks connect loci on two differ</p>
<p>** WORK IN PROGRESS **</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Thomas Hackl, Markus J. Ankenbrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
